<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Evgenii Tretiakov" />


<title>Ageing hypothalamic astrocytes</title>

<script src="site_libs/header-attrs-2.18/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">heteroAstrocytes</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="hfd.html">HFD</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Ageing hypothalamic astrocytes</h1>
<h4 class="author">Evgenii Tretiakov</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-11-24
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>1_heteroAstrocytes/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R
Markdown file created these results, you’ll want to first commit it to
the Git repo. If you’re still working on the analysis, you can ignore
this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220624code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20220624)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220624code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220624)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrong1cfad7d">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong> 1cfad7d
</a>
</p>
</div>
<div id="strongRepositoryversionstrong1cfad7d"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version 1cfad7d.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.Rhistory
    Ignored:    analysis/figure/
    Ignored:    data/comb_astrocytes.bak.h5Seurat
    Ignored:    data/comb_astrocytes.bak.h5ad
    Ignored:    data/deng2019_arc_chow.h5Seurat
    Ignored:    data/deng2019_arc_chow.h5ad
    Ignored:    data/deng2019_arc_chow_astrocytes.h5Seurat
    Ignored:    data/deng2019_arc_chow_astrocytes.h5ad
    Ignored:    data/deng2019_arc_chow_astrocytes_fin.h5Seurat
    Ignored:    data/deng2019_arc_chow_astrocytes_fin.h5ad
    Ignored:    data/deng2019_arc_chow_clusters.h5Seurat
    Ignored:    data/deng2019_arc_chow_clusters.h5ad
    Ignored:    data/deng2019_arc_hfd.h5Seurat
    Ignored:    data/deng2019_arc_hfd.h5ad
    Ignored:    data/deng2019_arc_hfd_astrocytes.h5Seurat
    Ignored:    data/deng2019_arc_hfd_astrocytes.h5ad
    Ignored:    data/deng2019_arc_hfd_astrocytes_fin.h5Seurat
    Ignored:    data/deng2019_arc_hfd_astrocytes_fin.h5ad
    Ignored:    data/deng2019_arc_hfd_clusters.h5Seurat
    Ignored:    data/deng2019_arc_hfd_clusters.h5ad
    Ignored:    data/deng2019_arc_refi_astrocytes.h5Seurat
    Ignored:    data/deng2019_arc_refi_astrocytes.h5ad
    Ignored:    data/deng_2020_arc_chow.h5Seurat
    Ignored:    data/deng_2020_arc_chow.h5ad
    Ignored:    data/deng_2020_arc_chow_astro-npy-pomc.h5Seurat
    Ignored:    data/deng_2020_arc_chow_astro-npy-pomc.h5ad
    Ignored:    data/deng_2020_arc_chow_astrocytes.h5Seurat
    Ignored:    data/deng_2020_arc_chow_astrocytes.h5ad
    Ignored:    data/deng_2020_arc_chow_astrocytes_fin.h5Seurat
    Ignored:    data/deng_2020_arc_chow_astrocytes_fin.h5ad
    Ignored:    data/deng_2020_arc_chow_clusters.h5Seurat
    Ignored:    data/deng_2020_arc_chow_clusters.h5ad
    Ignored:    data/deng_2020_arc_hfd.h5Seurat
    Ignored:    data/deng_2020_arc_hfd.h5ad
    Ignored:    data/deng_2020_arc_hfd_astrocytes.h5Seurat
    Ignored:    data/deng_2020_arc_hfd_astrocytes.h5ad
    Ignored:    data/deng_2020_arc_hfd_astrocytes_fin.h5Seurat
    Ignored:    data/deng_2020_arc_hfd_astrocytes_fin.h5ad
    Ignored:    data/deng_2020_arc_hfd_clusters.h5Seurat
    Ignored:    data/deng_2020_arc_hfd_clusters.h5ad
    Ignored:    data/deng_2020_arc_refi_astrocytes.h5Seurat
    Ignored:    data/deng_2020_arc_refi_astrocytes.h5ad
    Ignored:    output/figures/

Untracked files:
    Untracked:  analysis/.ipynb_checkpoints/
    Untracked:  analysis/2022-11-16_pr.Rmd
    Untracked:  analysis/2022-11-16_pr2.Rmd
    Untracked:  analysis/2022-11-23_pr.Rmd
    Untracked:  analysis/deng2020_norm_ligrec.pickle
    Untracked:  analysis/normalised.ipynb
    Untracked:  output/tables/ageing_astr_k9-mrk_logreg-sct.csv
    Untracked:  output/tables/ageing_astr_k9-mrk_roc-sct.csv
    Untracked:  output/tables/astr_k17-mrk_wilcox-sct.csv
    Untracked:  output/tables/astr_reg-mrk_logreg-sct-notintegrated.csv
    Untracked:  output/tables/astr_reg-mrk_roc-sct-notintegrated.csv
    Untracked:  output/tables/genes-meta_astroMap-all.tsv
    Untracked:  output/tables/genes_astroMap-all.tsv

Unstaged changes:
    Modified:   analysis/hfd.Rmd
    Modified:   code/genes.R
    Modified:   output/tables/arc_ast.csv

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with
<code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<pre class="r"><code>genes.embed &lt;- c(&quot;Ndrg2&quot;, &quot;Slc1a3&quot;, &quot;Gfap&quot;, 
                 &quot;S100b&quot;, &quot;Agt&quot;, &quot;Slc6a11&quot;,
                 &quot;Aqp4&quot;, &quot;Nfia&quot;, &quot;Aldoc&quot;, 
                 &quot;Apoe&quot;, &quot;Aldh1l1&quot;, &quot;Sox9&quot;, 
                 &quot;Fgfr3&quot;, &quot;Egfr&quot;, &quot;Ntrk2&quot;)
srt &lt;- readRDS(here(raw_dir, &quot;GSE188646_hypo.integrated.final.20210719.RDS&quot;))</code></pre>
<pre class="r"><code>skim(srt@meta.data)</code></pre>
<pre><code>── Data Summary ────────────────────────
                           Values       
Name                       srt@meta.data
Number of rows             40064        
Number of columns          8            
_______________________                 
Column type frequency:                  
  character                2            
  factor                   3            
  numeric                  3            
________________________                
Group variables            None         

── Variable type: character ────────────────────────────────────────────────────
  skim_variable n_missing complete_rate min max empty n_unique whitespace
1 orig.ident            0             1   6   7     0        8          0
2 stim                  0             1   4   5     0        2          0

── Variable type: factor ───────────────────────────────────────────────────────
  skim_variable          n_missing complete_rate ordered n_unique
1 integrated_snn_res.1.5         0             1 FALSE         59
2 seurat_clusters                0             1 FALSE         59
3 group                          0             1 FALSE          9
  top_counts                                 
1 0: 2418, 1: 2213, 2: 1911, 3: 1812         
2 0: 2418, 1: 2213, 2: 1911, 3: 1812         
3 Neu: 26940, Oli: 5027, Ast: 4494, Mic: 1218

── Variable type: numeric ──────────────────────────────────────────────────────
  skim_variable n_missing complete_rate    mean      sd  p0      p25      p50
1 nCount_RNA            0             1 5379.   5811.   498 1646     3479    
2 nFeature_RNA          0             1 2258.   1510.   207 1055     1954.   
3 percent.mt            0             1    1.61    1.91   0    0.202    0.900
      p75     p100 hist 
1 5935.   55149    ▇▁▁▁▁
2 2852     7495    ▇▇▂▂▁
3    2.30     9.99 ▇▂▁▁▁</code></pre>
<pre class="r"><code>srt</code></pre>
<pre><code>An object of class Seurat 
27135 features across 40064 samples within 2 assays 
Active assay: integrated (2000 features, 2000 variable features)
 1 other assay present: RNA
 2 dimensional reductions calculated: pca, umap</code></pre>
<pre class="r"><code>DimPlot(srt, reduction = &quot;umap&quot;, 
                         group.by = &quot;orig.ident&quot;, 
                         label = TRUE, repel = TRUE) + NoLegend()</code></pre>
<p><img src="figure/2022-11-23_pr.Rmd/pl-all-orginally-integrated-astrocytes-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plEmbOrig &lt;- DimPlot(srt, reduction = &quot;umap&quot;,
                     group.by = &quot;group&quot;, 
                     label = TRUE, repel = TRUE) + NoLegend()
plEmbOrig</code></pre>
<p><img src="figure/2022-11-23_pr.Rmd/pl-all-orginally-integrated-astrocytes-2.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>if (!file.exists(here(raw_dir, &quot;GSE188646.h5Seurat&quot;))) {
  SaveH5Seurat(srt, 
               filename = here(raw_dir, &quot;GSE188646.h5Seurat&quot;))
  Convert(here(raw_dir, &quot;GSE188646.h5Seurat&quot;), 
          dest = &quot;h5ad&quot;)
}

Idents(srt) &lt;- &quot;group&quot;
astro &lt;- 
    srt |&gt;  
    subset(idents = c(&quot;Astrocyte&quot;))
if (!file.exists(here(raw_dir, &quot;astro.h5Seurat&quot;))) {
  SaveH5Seurat(astro, 
               filename = here(raw_dir, &quot;astro.h5Seurat&quot;))
  Convert(here(raw_dir, &quot;astro.h5Seurat&quot;), 
          dest = &quot;h5ad&quot;)
}

gc()</code></pre>
<pre><code>            used   (Mb) gc trigger   (Mb)  max used   (Mb)
Ncells   8175944  436.7   14670914  783.6  14670914  783.6
Vcells 510276597 3893.2  666118601 5082.1 523684594 3995.4</code></pre>
<pre class="r"><code>DefaultAssay(object = astro) &lt;- &quot;RNA&quot;
astro &lt;-
  SCTransform(astro, vst.flavor = &quot;v2&quot;,
              # vars.to.regress = &quot;percent_mt&quot;,
              method = &quot;glmGamPoi&quot;,
              variable.features.n = 5000,
              return.only.var.genes = FALSE,
              seed.use = reseed,
              verbose = FALSE) |&gt; 
  RunPCA(seed.use = reseed,
         verbose = FALSE) |&gt; 
  RunUMAP(reduction = &quot;pca&quot;, 
          densmap = TRUE, dims = 1:50, 
          seed.use = reseed, verbose = FALSE)
plEmbRawBatch &lt;- DimPlot(astro, reduction = &quot;umap&quot;, 
                         group.by = &quot;orig.ident&quot;, 
                         label = TRUE, repel = TRUE) + NoLegend()
plEmbRawReg &lt;- DimPlot(astro, reduction = &quot;umap&quot;, 
                       group.by = &quot;stim&quot;,
                       label = TRUE, repel = TRUE) + NoLegend()
plEmbRawBatch + plEmbRawReg</code></pre>
<p><img src="figure/2022-11-23_pr.Rmd/pl-all-not-integrated-astrocytes-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>Integration of astrocytes</p>
<pre class="r"><code>srt_list &lt;- Seurat::SplitObject(astro, split.by = &quot;orig.ident&quot;)
srt_list &lt;- lapply(X = srt_list,
                   vst.flavor = &quot;v2&quot;,
                   FUN = SCTransform,
                   # vars.to.regress = &quot;percent_mt&quot;,
                   method = &quot;glmGamPoi&quot;,
                   variable.features.n = 5000,
                   return.only.var.genes = FALSE,
                   seed.use = reseed,
                   verbose = FALSE)
features &lt;- SelectIntegrationFeatures(object.list = srt_list, nfeatures = 3000, verbose = FALSE)</code></pre>
<pre class="r"><code>npcs &lt;- 30
srt_list &lt;- lapply(X = srt_list, FUN = RunPCA, features = features, npcs = npcs, seed.use = reseed, verbose = FALSE)

srt_list &lt;- PrepSCTIntegration(object.list = srt_list, anchor.features = features)

anchors &lt;- FindIntegrationAnchors(
  object.list = srt_list,
  normalization.method = &quot;SCT&quot;,
  anchor.features = features,
  reduction = &quot;rpca&quot;,
  dims = 1:30,
  k.anchor = 20,
  verbose = FALSE
)

combined_sct &lt;- IntegrateData(
  anchorset = anchors,
  normalization.method = &quot;SCT&quot;,
  dims = 1:30,
  verbose = FALSE
)

metadata = combined_sct@meta.data
rownames(metadata) = colnames(combined_sct)
ref.labels = metadata$stim
combined_sct &lt;-
  RunPCA(combined_sct, npcs = npcs, 
         seed.use = reseed, verbose = FALSE)
combined_sct</code></pre>
<pre><code>An object of class Seurat 
44112 features across 4494 samples within 3 assays 
Active assay: integrated (3000 features, 3000 variable features)
 2 other assays present: RNA, SCT
 1 dimensional reduction calculated: pca</code></pre>
<pre class="r"><code>combined_sct &lt;- combined_sct |&gt; 
  RunUMAP(reduction = &quot;pca&quot;, return.model = TRUE,
          densmap = TRUE, dims = 1:npcs, 
          seed.use = reseed, verbose = FALSE) |&gt; 
  FindNeighbors(reduction = &quot;pca&quot;, dims = 1:npcs, verbose = FALSE)

resolutions &lt;-
  modularity_event_sampling(
    A = combined_sct@graphs$integrated_snn,
    n.res = 10,
    gamma.min = 0.1,
    gamma.max = 2.5
  ) # sample based on the similarity matrix

# clustering using Suerat
combined_sct &lt;- combined_sct |&gt; 
  FindClusters(algorithm = 4, method = &quot;igraph&quot;,
               resolution = resolutions, random.seed = reseed,
               verbose = FALSE)

# initial cluster tree from Seurat flat clustering
plot_clustree(
  labelmat = combined_sct@meta.data,
  prefix = &#39;integrated_snn_res.&#39;,
  ref.labels = ref.labels,
  plot.ref = FALSE
)</code></pre>
<p><img src="figure/2022-11-23_pr.Rmd/pl-clustree-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Adjusted Multiresolution Rand Index (AMRI)
ks.flat &lt;-  apply(
  out$labelmat.flat,
  2,
  FUN = function(x)
    length(unique(x))
)
ks.mrtree &lt;-  apply(
  out$labelmat.mrtree,
  2,
  FUN = function(x)
    length(unique(x))
)
amri.flat &lt;-  sapply(1:ncol(out$labelmat.flat), function(i)
  AMRI(out$labelmat.flat[, i], ref.labels)$amri)
amri.flat &lt;-  aggregate(amri.flat, by = list(k = ks.flat), FUN = mean)
amri.recon &lt;-  sapply(1:ncol(out$labelmat.mrtree), function(i)
  AMRI(out$labelmat.mrtree[, i], ref.labels)$amri)

df &lt;-  rbind(
  data.frame(
    k = amri.flat$k,
    amri = amri.flat$x,
    method = &#39;Seurat flat&#39;
  ),
  data.frame(k = ks.mrtree, amri = amri.recon, method = &#39;MRtree&#39;)
)
ggplot2::ggplot(data = df, aes(x = k, y = amri, color = method)) + geom_line() + theme_bw()</code></pre>
<p><img src="figure/2022-11-23_pr.Rmd/pl-clustering-amri-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>stab.out &lt;- stability_plot(out)
stab.out$plot</code></pre>
<p><img src="figure/2022-11-23_pr.Rmd/pl-clustering-resolution-1.png" width="480" style="display: block; margin: auto;" /></p>
<p>Plot by source</p>
<pre class="r"><code>plEmbCombBatch &lt;- DimPlot(combined_sct, reduction = &quot;umap&quot;,
                          group.by = &quot;orig.ident&quot;,
                          label = TRUE, repel = TRUE) + NoLegend()
plEmbCombReg &lt;- DimPlot(combined_sct, reduction = &quot;umap&quot;,
                        group.by = &quot;stim&quot;,
                        label = TRUE, repel = TRUE) + NoLegend()
plEmbCombBatch + plEmbCombReg</code></pre>
<p><img src="figure/2022-11-23_pr.Rmd/pl-all-integrated-astrocytes-nucleus-1.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>stab.out$df</code></pre>
<pre><code>  resolution       ari
1          5 0.1936614
2          6 0.9171808
3          8 0.9601062
4          9 0.9746733
5         12 0.8306028
6         13 0.8139924
7         15 0.8242993
8         17 0.9087655
9         21 0.8887843</code></pre>
<pre class="r"><code>combined_sct$k_tree &lt;- as.factor(out$labelmat.mrtree[, &quot;K17&quot;])
p1 &lt;- DimPlot(combined_sct, label = T, repel = T) + ggtitle(&quot;Unsupervised clustering&quot;) + NoLegend()
p2 &lt;- DimPlot(combined_sct, label = T, repel = T, group.by = &quot;k_tree&quot;) + ggtitle(&quot;MRTree&quot;) + NoLegend()

p1 | p2</code></pre>
<p><img src="figure/2022-11-23_pr.Rmd/pl-clustering-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>library(schex)
combined_sct &lt;- make_hexbin(combined_sct, nbins = 60, dimension_reduction = &quot;UMAP&quot;)
plot_hexbin_density(combined_sct)</code></pre>
<p><img src="figure/2022-11-23_pr.Rmd/pl-schex-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot_hexbin_meta(combined_sct, col = &quot;nCount_RNA&quot;, action = &quot;median&quot;)</code></pre>
<p><img src="figure/2022-11-23_pr.Rmd/pl-schex-2.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot_hexbin_meta(combined_sct, col = &quot;k_tree&quot;, action = &quot;majority&quot;)</code></pre>
<p><img src="figure/2022-11-23_pr.Rmd/pl-schex-3.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>label_df &lt;- make_hexbin_label(combined_sct, col = &quot;k_tree&quot;)
pp &lt;- plot_hexbin_meta(combined_sct, col = &quot;k_tree&quot;, action = &quot;majority&quot;)
pp + ggrepel::geom_label_repel(data = label_df, aes(x = x, y = y, label = label), colour = &quot;black&quot;, 
    label.size = NA, fill = NA)</code></pre>
<p><img src="figure/2022-11-23_pr.Rmd/pl-schex-k-tree-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>reg_df &lt;- make_hexbin_label(combined_sct, 
                            col = &quot;orig.ident&quot;)
pp2 &lt;- plot_hexbin_meta(combined_sct,
                        col = &quot;orig.ident&quot;,
                        action = &quot;majority&quot;)
pp2 + ggrepel::geom_label_repel(data = reg_df, 
                                aes(x = x, y = y,
                                    label = label), 
                                colour = &quot;black&quot;,
                                label.size = NA, fill = NA)</code></pre>
<p><img src="figure/2022-11-23_pr.Rmd/pl-schex-regions-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>Idents(combined_sct) &lt;- &quot;k_tree&quot;
combined_sct &lt;- PrepSCTFindMarkers(combined_sct)

srt.markers.roc &lt;- FindAllMarkers(combined_sct, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25, test.use = &quot;roc&quot;)
readr::write_csv(srt.markers.roc, 
                 here(tables_dir,
                      &#39;ageing_astr_k9-mrk_roc-sct.csv&#39;))

srt.markers.roc %&gt;%
    group_by(cluster) %&gt;%
    slice_max(n = 3, order_by = avg_log2FC)</code></pre>
<pre><code># A tibble: 40 × 8
# Groups:   cluster [15]
   myAUC avg_diff power avg_log2FC pct.1 pct.2 cluster gene         
   &lt;dbl&gt;    &lt;dbl&gt; &lt;dbl&gt;      &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;fct&gt;   &lt;chr&gt;        
 1 0.804    1.51  0.608       2.18 0.798 0.263 1       Aldh1a1      
 2 0.74     0.880 0.48        1.27 0.739 0.346 1       Gm44151      
 3 0.712    0.785 0.424       1.13 0.645 0.252 1       Egfl6        
 4 0.711    1.51  0.422       2.18 0.625 0.183 4       Sfta3-ps     
 5 0.742    1.36  0.484       1.96 0.701 0.213 4       Pdzph1       
 6 0.711    1.25  0.422       1.80 0.705 0.278 4       Slit2        
 7 0.823    1.81  0.646       2.61 0.851 0.377 5       Rpl26        
 8 0.808    1.79  0.616       2.59 0.858 0.401 5       Fth1         
 9 0.822    1.66  0.644       2.39 0.861 0.317 5       Rpl23a       
10 0.802    0.902 0.604       1.30 0.79  0.327 6       1700034P13Rik
# … with 30 more rows</code></pre>
<pre class="r"><code>srt.markers.lr &lt;- FindAllMarkers(combined_sct, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25, test.use = &quot;LR&quot;)
readr::write_csv(srt.markers.lr, 
                 here(tables_dir,
                      &#39;ageing_astr_k9-mrk_logreg-sct.csv&#39;))
srt.markers.lr %&gt;%
    group_by(cluster) %&gt;%
    slice_max(n = 3, order_by = avg_log2FC)</code></pre>
<pre><code># A tibble: 51 × 7
# Groups:   cluster [17]
       p_val avg_log2FC pct.1 pct.2 p_val_adj cluster gene         
       &lt;dbl&gt;      &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;     &lt;dbl&gt; &lt;fct&gt;   &lt;chr&gt;        
 1 6.20e-137       2.18 0.798 0.263 1.86e-133 1       Aldh1a1      
 2 1.78e- 57       1.76 0.604 0.222 5.35e- 54 1       Tafa1        
 3 3.89e- 29       1.35 0.581 0.305 1.17e- 25 1       Esrrg        
 4 5.31e- 10       1.47 0.374 0.261 1.59e-  6 2       Pacrg        
 5 4.36e-  9       1.42 0.34  0.231 1.31e-  5 2       Gm36037      
 6 4.92e-  9       1.25 0.435 0.305 1.48e-  5 2       Otx2os1      
 7 2.28e-  6       1.42 0.369 0.336 6.83e-  3 3       Frmpd1       
 8 3.12e-  3       1.20 0.239 0.259 1   e+  0 3       D130040H23Rik
 9 5.16e-  4       1.19 0.273 0.273 1   e+  0 3       Kansl3       
10 6.79e- 66       2.18 0.625 0.183 2.04e- 62 4       Sfta3-ps     
# … with 41 more rows</code></pre>
<pre class="r"><code>srt.markers.lr %&gt;%
    group_by(cluster) %&gt;%
    top_n(n = 10, wt = avg_log2FC) -&gt; top10
DoHeatmap(combined_sct, features = top10$gene, 
          size = 4, angle = 90) + NoLegend()</code></pre>
<p><img src="figure/2022-11-23_pr.Rmd/pl-heatmap-1.png" width="1344" style="display: block; margin: auto;" /></p>
<pre class="r"><code>library(Scillus)
plot_heatmap(
  dataset = combined_sct,
  markers = srt.markers.lr,
  sort_var = c(&quot;k_tree&quot;, &quot;orig.ident&quot;),
  anno_var = c(
    &quot;k_tree&quot;,
    &quot;orig.ident&quot;,
    &quot;stim&quot;,
    &quot;percent.mt&quot;,
    &quot;nFeature_RNA&quot;,
    &quot;nCount_RNA&quot;
  ),
  anno_colors = list(
    &quot;Set2&quot;,
    # RColorBrewer palette
    c(&quot;navy&quot;, &quot;red&quot;, &quot;orange&quot;, &quot;gold&quot;, &quot;darkorchid3&quot;, &quot;dodgerblue&quot;, &quot;forestgreen&quot;, &quot;orchid&quot;),
    # color vector
    c(&quot;navy&quot;, &quot;gold&quot;),
    &quot;Greens&quot;,
    c(&quot;blue&quot;, &quot;white&quot;, &quot;red&quot;),
    # Three-color gradient
    &quot;Reds&quot;
  )
)</code></pre>
<p><img src="figure/2022-11-23_pr.Rmd/pl-heatmap-logreg-1.png" width="1344" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot_heatmap(
  dataset = combined_sct,
  markers = srt.markers.roc,
  sort_var = c(&quot;k_tree&quot;, &quot;orig.ident&quot;),
  anno_var = c(
    &quot;k_tree&quot;,
    &quot;orig.ident&quot;,
    &quot;stim&quot;,
    &quot;percent.mt&quot;,
    &quot;nFeature_RNA&quot;,
    &quot;nCount_RNA&quot;
  ),
  anno_colors = list(
    &quot;Set2&quot;,
    # RColorBrewer palette
    c(&quot;navy&quot;, &quot;red&quot;, &quot;orange&quot;, &quot;gold&quot;, &quot;darkorchid3&quot;, &quot;dodgerblue&quot;, &quot;forestgreen&quot;, &quot;orchid&quot;),
    # color vector
    c(&quot;navy&quot;, &quot;gold&quot;),
    &quot;Greens&quot;,
    c(&quot;blue&quot;, &quot;white&quot;, &quot;red&quot;),
    # Three-color gradient
    &quot;Reds&quot;
  )
)</code></pre>
<p><img src="figure/2022-11-23_pr.Rmd/pl-heatmap-roc-1.png" width="1344" style="display: block; margin: auto;" /></p>
<pre class="r"><code>library(scCustomize)

gene_list_plot &lt;- 
  Extract_Top_Markers(marker_dataframe = srt.markers.roc,
                      num_genes = 3,
                      rank_by = &quot;avg_log2FC&quot;,
                      make_unique = TRUE,
                      named_vector = FALSE)

sample_colors &lt;- c(&quot;darkorchid3&quot;, &quot;gold&quot;)

# Create Plots
Stacked_VlnPlot(seurat_object = combined_sct, features = gene_list_plot, x_lab_rotate = TRUE,
    colors_use = sample_colors, split.by = &quot;stim&quot;)</code></pre>
<p><img src="figure/2022-11-23_pr.Rmd/pl-vln-meta-1.png" width="1344" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plEmbCombCMrk &lt;- FeaturePlot(
  combined_sct,
  features = genes.embed[genes.embed %in%
                           rownames(combined_sct@assays$SCT@data)],
  min.cutoff = &quot;q15&quot;, max.cutoff = &quot;q85&quot;,
  ncol = 4, order = TRUE
)
plEmbCombCMrk</code></pre>
<p><img src="figure/2022-11-23_pr.Rmd/pl-all-integrated-astrocytes-init-canon-markers-1.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plEmbCombCMrk2 &lt;- FeaturePlot(
  combined_sct,
  features = gene_list_plot,
  min.cutoff = &quot;q15&quot;, max.cutoff = &quot;q85&quot;,
  ncol = 4, order = TRUE
)
plEmbCombCMrk2</code></pre>
<p><img src="figure/2022-11-23_pr.Rmd/pl-all-integrated-astrocytes-init-markers-1.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>gene_list_plot2 &lt;- 
  Extract_Top_Markers(marker_dataframe = srt.markers.lr,
                      num_genes = 3,
                      rank_by = &quot;avg_log2FC&quot;,
                      make_unique = TRUE,
                      named_vector = FALSE)
plEmbCombCMrk3 &lt;- FeaturePlot(
  combined_sct,
  features = gene_list_plot2,
  min.cutoff = &quot;q15&quot;, max.cutoff = &quot;q85&quot;,
  ncol = 4, order = TRUE
)
plEmbCombCMrk3</code></pre>
<p><img src="figure/2022-11-23_pr.Rmd/pl-all-integrated-astrocytes-init-markers2-1.png" width="1152" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.2 Patched (2022-11-10 r83330)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 22.04.1 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3
LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] scCustomize_1.0.2           Scillus_0.5.0              
 [3] schex_1.12.0                shiny_1.7.3                
 [5] SingleCellExperiment_1.20.0 SummarizedExperiment_1.28.0
 [7] Biobase_2.58.0              GenomicRanges_1.50.1       
 [9] GenomeInfoDb_1.34.3         IRanges_2.32.0             
[11] S4Vectors_0.36.0            BiocGenerics_0.44.0        
[13] MatrixGenerics_1.10.0       matrixStats_0.63.0         
[15] ggraph_2.1.0                mrtree_0.0.0.9000          
[17] Nebulosa_1.8.0              swne_0.6.20                
[19] patchwork_1.1.2             UpSetR_1.4.0               
[21] glmGamPoi_1.10.0            sctransform_0.3.5          
[23] SeuratDisk_0.0.0.9020       SeuratWrappers_0.3.1       
[25] SeuratObject_4.1.3          Seurat_4.3.0               
[27] reticulate_1.26             kableExtra_1.3.4           
[29] future_1.29.0               skimr_2.1.4                
[31] magrittr_2.0.3              forcats_0.5.2              
[33] stringr_1.4.1               dplyr_1.0.10               
[35] purrr_0.3.5                 readr_2.1.3                
[37] tidyr_1.2.1                 tibble_3.1.8               
[39] ggplot2_3.4.0               tidyverse_1.3.2            
[41] here_1.0.1                  workflowr_1.7.0            

loaded via a namespace (and not attached):
  [1] rsvd_1.0.5              ica_1.0-3               svglite_2.1.0          
  [4] ps_1.7.2                foreach_1.5.2           lmtest_0.9-40          
  [7] rprojroot_2.0.3         crayon_1.5.2            MASS_7.3-58.1          
 [10] nlme_3.1-160            backports_1.4.1         reprex_2.0.2           
 [13] rlang_1.0.6             XVector_0.38.0          ROCR_1.0-11            
 [16] readxl_1.4.1            irlba_2.3.5.1           callr_3.7.3            
 [19] data.tree_1.0.0         rjson_0.2.21            bit64_4.0.5            
 [22] glue_1.6.2              vipor_0.4.5             parallel_4.2.2         
 [25] processx_3.8.0          spatstat.sparse_3.0-0   spatstat.geom_3.0-3    
 [28] haven_2.5.1             tidyselect_1.2.0        liger_2.0.1            
 [31] fitdistrplus_1.1-8      zoo_1.8-11              ggpubr_0.5.0           
 [34] formattable_0.2.1       xtable_1.8-4            evaluate_0.18          
 [37] cli_3.4.1               zlibbioc_1.44.0         rstudioapi_0.14        
 [40] miniUI_0.1.1.1          sp_1.5-1                whisker_0.4            
 [43] bslib_0.4.1             fastmatch_1.1-3         treeio_1.23.0          
 [46] maps_3.4.1              xfun_0.35               clue_0.3-63            
 [49] askpass_1.1             usedist_0.4.0           cluster_2.1.4          
 [52] tidygraph_1.2.2         clusterGeneration_1.3.7 expm_0.999-6           
 [55] SymSim_0.0.0.9000       ggrepel_0.9.2           ape_5.6-2              
 [58] listenv_0.8.0           dendextend_1.16.0       png_0.1-7              
 [61] withr_2.5.0             bitops_1.0-7            ggforce_0.4.1          
 [64] plyr_1.8.8              cellranger_1.1.0        pracma_2.4.2           
 [67] coda_0.19-4             pillar_1.8.1            GlobalOptions_0.1.2    
 [70] cachem_1.0.6            fs_1.5.2                scatterplot3d_0.3-42   
 [73] hdf5r_1.3.7             GetoptLong_1.0.5        paletteer_1.5.0        
 [76] vctrs_0.5.1             ellipsis_0.3.2          generics_0.1.3         
 [79] tools_4.2.2             beeswarm_0.4.0          entropy_1.3.1          
 [82] munsell_0.5.0           tweenr_2.0.2            proxy_0.4-27           
 [85] DelayedArray_0.24.0     fastmap_1.1.0           compiler_4.2.2         
 [88] abind_1.4-5             httpuv_1.6.6            ggimage_0.3.1          
 [91] plotly_4.10.1           GenomeInfoDbData_1.2.9  gridExtra_2.3          
 [94] colorway_0.2.0          lattice_0.20-45         deldir_1.0-6           
 [97] snow_0.4-4              utf8_1.2.2              later_1.3.0            
[100] jsonlite_1.8.3          concaveman_1.1.0        scales_1.2.1           
[103] carData_3.0-5           tidytree_0.4.1          pbapply_1.6-0          
[106] lazyeval_0.2.2          promises_1.2.0.1        car_3.1-1              
[109] doParallel_1.0.17       R.utils_2.12.2          goftest_1.2-3          
[112] spatstat.utils_3.0-1    checkmate_2.1.0         rmarkdown_2.18         
[115] cowplot_1.1.1           webshot_0.5.4           Rtsne_0.16             
[118] uwot_0.1.14             igraph_1.3.5            survival_3.4-0         
[121] numDeriv_2016.8-1.1     yaml_2.3.6              plotrix_3.8-2          
[124] systemfonts_1.0.4       htmltools_0.5.3         graphlayouts_0.8.3     
[127] quadprog_1.5-8          viridisLite_0.4.1       digest_0.6.30          
[130] assertthat_0.2.1        mime_0.12               repr_1.1.4             
[133] yulab.utils_0.0.5       future.apply_1.10.0     ggmin_0.0.0.9000       
[136] remotes_2.4.2           data.table_1.14.6       R.oo_1.25.0            
[139] splines_4.2.2           labeling_0.4.2          rematch2_2.1.2         
[142] Cairo_1.6-0             googledrive_2.0.0       RCurl_1.98-1.9         
[145] broom_1.0.1             ks_1.13.5               hms_1.1.2              
[148] modelr_0.1.10           colorspace_2.0-3        base64enc_0.1-3        
[151] BiocManager_1.30.19     mnormt_2.1.1            ggbeeswarm_0.6.0       
[154] shape_1.4.6             aplot_0.1.8             ggrastr_1.0.1          
[157] sass_0.4.3              Rcpp_1.0.9              mclust_6.0.0           
[160] RANN_2.6.1              circlize_0.4.15         mvtnorm_1.1-3          
[163] NNLM_0.4.4              fansi_1.0.3             tzdb_0.3.0             
[166] parallelly_1.32.1       R6_2.5.1                grid_4.2.2             
[169] ggridges_0.5.4          lifecycle_1.0.3         phytools_1.2-0         
[172] ggsignif_0.6.4          googlesheets4_1.0.1     leiden_0.4.3           
[175] phangorn_2.10.0         jquerylib_0.1.4         snakecase_0.11.0       
[178] Matrix_1.5-3            RcppAnnoy_0.0.20        RColorBrewer_1.1-3     
[181] iterators_1.0.14        spatstat.explore_3.0-5  htmlwidgets_1.5.4      
[184] umap_0.2.9.0            polyclip_1.10-4         timechange_0.1.1       
[187] gridGraphics_0.5-1      optimParallel_1.0-2     ComplexHeatmap_2.14.0  
[190] rvest_1.0.3             mgcv_1.8-41             globals_0.16.2         
[193] openssl_2.0.4           spatstat.random_3.0-1   progressr_0.11.0       
[196] codetools_0.2-18        lubridate_1.9.0         FNN_1.1.3.1            
[199] getPass_0.2-2           dbplyr_2.2.1            RSpectra_0.16-1        
[202] R.methodsS3_1.8.2       gtable_0.3.1            DBI_1.1.3              
[205] git2r_0.30.1            highr_0.9               ggfun_0.0.9            
[208] tensor_1.5              httr_1.4.4              KernSmooth_2.23-20     
[211] vroom_1.6.0             stringi_1.7.8           reshape2_1.4.4         
[214] farver_2.1.1            viridis_0.6.2           hexbin_1.28.2          
[217] magick_2.7.3            ggtree_3.7.1.002        xml2_1.3.3             
[220] combinat_0.0-8          ggplotify_0.1.0         scattermore_0.8        
[223] bit_4.0.5               clustree_0.5.0          spatstat.data_3.0-0    
[226] janitor_2.1.0           pkgconfig_2.0.3         gargle_1.2.1           
[229] ggprism_1.0.4           rstatix_0.7.1           knitr_1.41             </code></pre>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
