<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Evgenii Tretiakov" />


<title>Ageing hypothalamic astrocytes to separate nuclei transfer</title>

<script src="site_libs/header-attrs-2.18/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">heteroAstrocytes</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="hfd.html">HFD</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Ageing hypothalamic astrocytes to separate
nuclei transfer</h1>
<h4 class="author">Evgenii Tretiakov</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-12-16
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>1_heteroAstrocytes/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R
Markdown file created these results, you’ll want to first commit it to
the Git repo. If you’re still working on the analysis, you can ignore
this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220624code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20220624)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220624code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220624)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrong1cfad7d">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong> 1cfad7d
</a>
</p>
</div>
<div id="strongRepositoryversionstrong1cfad7d"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version 1cfad7d.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.Rhistory
    Ignored:    analysis/figure/
    Ignored:    data/comb_astrocytes.bak.h5Seurat
    Ignored:    data/comb_astrocytes.bak.h5ad
    Ignored:    data/deng2019_arc_chow.h5Seurat
    Ignored:    data/deng2019_arc_chow.h5ad
    Ignored:    data/deng2019_arc_chow_astrocytes.h5Seurat
    Ignored:    data/deng2019_arc_chow_astrocytes.h5ad
    Ignored:    data/deng2019_arc_chow_astrocytes_fin.h5Seurat
    Ignored:    data/deng2019_arc_chow_astrocytes_fin.h5ad
    Ignored:    data/deng2019_arc_chow_clusters.h5Seurat
    Ignored:    data/deng2019_arc_chow_clusters.h5ad
    Ignored:    data/deng2019_arc_hfd.h5Seurat
    Ignored:    data/deng2019_arc_hfd.h5ad
    Ignored:    data/deng2019_arc_hfd_astrocytes.h5Seurat
    Ignored:    data/deng2019_arc_hfd_astrocytes.h5ad
    Ignored:    data/deng2019_arc_hfd_astrocytes_fin.h5Seurat
    Ignored:    data/deng2019_arc_hfd_astrocytes_fin.h5ad
    Ignored:    data/deng2019_arc_hfd_clusters.h5Seurat
    Ignored:    data/deng2019_arc_hfd_clusters.h5ad
    Ignored:    data/deng2019_arc_refi_astrocytes.h5Seurat
    Ignored:    data/deng2019_arc_refi_astrocytes.h5ad
    Ignored:    data/deng_2020_arc_chow.h5Seurat
    Ignored:    data/deng_2020_arc_chow.h5ad
    Ignored:    data/deng_2020_arc_chow_astro-npy-pomc.h5Seurat
    Ignored:    data/deng_2020_arc_chow_astro-npy-pomc.h5ad
    Ignored:    data/deng_2020_arc_chow_astrocytes.h5Seurat
    Ignored:    data/deng_2020_arc_chow_astrocytes.h5ad
    Ignored:    data/deng_2020_arc_chow_astrocytes_fin.h5Seurat
    Ignored:    data/deng_2020_arc_chow_astrocytes_fin.h5ad
    Ignored:    data/deng_2020_arc_chow_clusters.h5Seurat
    Ignored:    data/deng_2020_arc_chow_clusters.h5ad
    Ignored:    data/deng_2020_arc_hfd.h5Seurat
    Ignored:    data/deng_2020_arc_hfd.h5ad
    Ignored:    data/deng_2020_arc_hfd_astrocytes.h5Seurat
    Ignored:    data/deng_2020_arc_hfd_astrocytes.h5ad
    Ignored:    data/deng_2020_arc_hfd_astrocytes_fin.h5Seurat
    Ignored:    data/deng_2020_arc_hfd_astrocytes_fin.h5ad
    Ignored:    data/deng_2020_arc_hfd_clusters.h5Seurat
    Ignored:    data/deng_2020_arc_hfd_clusters.h5ad
    Ignored:    data/deng_2020_arc_refi_astrocytes.h5Seurat
    Ignored:    data/deng_2020_arc_refi_astrocytes.h5ad

Untracked files:
    Untracked:  analysis/.ipynb_checkpoints/
    Untracked:  analysis/2022-11-16_pr.Rmd
    Untracked:  analysis/2022-11-16_pr2.Rmd
    Untracked:  analysis/2022-11-23_pr.Rmd
    Untracked:  analysis/2022-11-24_pr.Rmd
    Untracked:  analysis/2022-11-25_pr.qmd
    Untracked:  analysis/2022-12-12_pr.Rmd
    Untracked:  analysis/deng2020_norm_ligrec.pickle
    Untracked:  analysis/normalised.ipynb
    Untracked:  data/astrocyte_enriched_genes_shared_by_mice.tsv
    Untracked:  data/housekeeping_mouse.tsv
    Untracked:  data/top_astrocytes_progenitor_cells_genes_by_humans.tsv
    Untracked:  data/top_mature_astrocyte_genes_by_humans.tsv
    Untracked:  output/figures/
    Untracked:  output/tables/ageing_astr_k9-mrk_logreg-sct.csv
    Untracked:  output/tables/ageing_astr_k9-mrk_roc-sct.csv
    Untracked:  output/tables/ageing_astr_recon-fix-mrk_logreg-sct.csv
    Untracked:  output/tables/ageing_astr_recon-fix-mrk_roc-sct.csv
    Untracked:  output/tables/ageing_astr_recon-mrk_logreg-sct.csv
    Untracked:  output/tables/ageing_astr_recon-mrk_roc-sct.csv
    Untracked:  output/tables/astr_k17-mrk_wilcox-sct.csv
    Untracked:  output/tables/astr_reg-mrk_logreg-sct-notintegrated.csv
    Untracked:  output/tables/astr_reg-mrk_roc-sct-notintegrated.csv
    Untracked:  output/tables/genes-meta_astroMap-all.tsv
    Untracked:  output/tables/genes_astroMap-all.tsv
    Untracked:  output/tables/novoseq-ageing_astr_recon-fix-mrk_logreg-sct.csv
    Untracked:  output/tables/novoseq-ageing_astr_recon-fix-mrk_mast-sct.csv
    Untracked:  output/tables/novoseq-ageing_astr_recon-fix-mrk_roc-sct.csv

Unstaged changes:
    Modified:   analysis/hfd.Rmd
    Modified:   code/genes.R
    Modified:   output/tables/arc_ast.csv

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with
<code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<pre class="r"><code>genes.embed &lt;- c(&quot;Ndrg2&quot;, &quot;Slc1a3&quot;, &quot;Gfap&quot;, 
                 &quot;S100b&quot;, &quot;Agt&quot;, &quot;Slc6a11&quot;,
                 &quot;Aqp4&quot;, &quot;Nfia&quot;, &quot;Aldoc&quot;, 
                 &quot;Apoe&quot;, &quot;Aldh1l1&quot;, &quot;Sox9&quot;, 
                 &quot;Fgfr3&quot;, &quot;Egfr&quot;, &quot;Ntrk2&quot;, &quot;Gja1&quot;)
genes.nature &lt;- c(&quot;Ndrg2&quot;, &quot;Aqp4&quot;, &quot;Gja1&quot;, 
                  &quot;Slc17a9&quot;, &quot;Slc17a7&quot;, &quot;Slc17a6&quot;, 
                  &quot;Slc16a1&quot;, &quot;Slc16a7&quot;, &quot;Slc16a3&quot;, 
                  &quot;Panx1&quot;, &quot;P2rx7&quot;, &quot;Srr&quot;, 
                  &quot;Dao&quot;, &quot;Vamp2&quot;, &quot;Gab1&quot;, 
                  &quot;Slc18a1&quot;, &quot;Slc18a2&quot;, &quot;Slc18a3&quot;, 
                  &quot;Slc17a5&quot;, &quot;Osmr&quot;, &quot;S100a6&quot;, 
                  &quot;Ogn&quot;, &quot;Itih5&quot;, &quot;Rdh10&quot;, 
                  &quot;Fst&quot;, &quot;1500015O10Rik&quot;, &quot;Rnf13&quot;, 
                  &quot;A2m&quot;, &quot;Rsph9&quot;, &quot;Galnt16&quot;, 
                  &quot;Rad23b&quot;, &quot;Tgfbr2&quot;, &quot;Ppp1r15a&quot;,
                  &quot;Mlc1&quot;, &quot;Slc6a11&quot;, &quot;Slc1a3&quot;, &quot;S100b&quot;,
                  &quot;S100b&quot;, &quot;Fabp7&quot;, &quot;Gab1&quot;, &quot;Igfbp5&quot;,
                  &quot;Hopx&quot;, &quot;Igsf1&quot;, &quot;Tgfb2&quot;, &quot;2810459M11Rik&quot;,
                  &quot;Rnf13&quot;, &quot;Itih5&quot;, &quot;Slc1a2&quot;, &quot;Cd59a&quot;,
                  &quot;Vim&quot;, &quot;Slc7a10&quot;, &quot;Fos&quot;)

genes.jj &lt;- c(
  &quot;Acsbg1&quot;, &quot;Acsl3&quot;, &quot;Actb&quot;, &quot;Agt&quot;, &quot;Aldh1l1&quot;, &quot;Aldoc&quot;, &quot;Apoe&quot;,
  &quot;Aqp4&quot;, &quot;Atp1a2&quot;, &quot;Bmpr1b&quot;, &quot;Cbs&quot;, &quot;Ckb&quot;, &quot;Clu&quot;, &quot;Cnx43&quot;,
  &quot;Cpe&quot;, &quot;Cst3&quot;, &quot;Cth&quot;, &quot;Cyp4f14&quot;, &quot;Dbi&quot;, &quot;Dbx2&quot;, &quot;Dctd&quot;,
  &quot;Dio2&quot;, &quot;Ednrb&quot;, &quot;Fgfr3&quot;, &quot;Gabbr1&quot;, &quot;Gabbr2&quot;, &quot;Gjb6&quot;, &quot;Gli3&quot;,
  &quot;Gm266&quot;, &quot;Gpr37l1&quot;, &quot;Grhl1&quot;, &quot;Grm3&quot;, &quot;Gs&quot;, &quot;Gucy2c&quot;, &quot;Hapln1&quot;,
  &quot;Hes5&quot;, &quot;Heyl&quot;, &quot;Hgf&quot;, &quot;Id2&quot;, &quot;Itih3&quot;, &quot;Lars2&quot;, &quot;Lcat&quot;,
  &quot;Ldhb&quot;, &quot;Malat1&quot;, &quot;Mc3r&quot;, &quot;Mfge8&quot;, &quot;Mlc1&quot;, &quot;Mmd2&quot;, &quot;Mt1&quot;,
  &quot;Mt2&quot;, &quot;Ndrg2&quot;, &quot;Nfia&quot;, &quot;Npas3&quot;, &quot;Nrxn1&quot;, &quot;Ntrk2&quot;, &quot;Ntsr2&quot;,
  &quot;Olx1&quot;, &quot;Otx2&quot;, &quot;Pax6&quot;, &quot;Pbxip1&quot;, &quot;Phkg1&quot;, &quot;Pla2g3&quot;, &quot;Pla2g7&quot;,
  &quot;Plcd4&quot;, &quot;Plce1&quot;, &quot;Plp1&quot;, &quot;Ppap2b&quot;, &quot;Ppia&quot;, &quot;Prodh&quot;, &quot;Ptprz1&quot;,
  &quot;Rfx4&quot;, &quot;Rpl41&quot;, &quot;S1pr1&quot;, &quot;Scd2&quot;, &quot;Serpinb1c&quot;, &quot;Slc19a3&quot;, &quot;Slc1a2&quot;,
  &quot;Slc1a3&quot;, &quot;Slc39a12&quot;, &quot;Slc4a4&quot;, &quot;Slc6a11&quot;, &quot;Son&quot;, &quot;Sox2&quot;, &quot;Sox9&quot;,
  &quot;Sparc&quot;, &quot;Sparcl1&quot;, &quot;Tlr3&quot;
)

genes.anatomy.jj &lt;- c(&quot;Gfap&quot;, &quot;Meg3&quot;, &quot;Snhg11&quot;, &quot;Ttc3&quot;, &quot;Cst3&quot;, &quot;Sparcl1&quot;,
                      &quot;Ndrg2&quot;, &quot;Cnx43&quot;, &quot;Agt&quot;, &quot;S100b&quot;, &quot;Aldh1l1&quot;, &quot;Lxn&quot;,
                      &quot;Aqp4&quot;, &quot;Slc1a2&quot;, &quot;Fabp7&quot;, &quot;Glul&quot;, &quot;Olig2&quot;)

# public resources:
housekeeping_mouse &lt;-
  read_lines(file = here(data_dir, &#39;housekeeping_mouse.tsv&#39;))
astroenriched_mouse &lt;-
  read_lines(file = here(data_dir, &#39;astrocyte_enriched_genes_shared_by_mice.tsv&#39;))

astroprogenitor_humans &lt;- 
  read_lines(file = here(data_dir, &#39;top_astrocytes_progenitor_cells_genes_by_humans.tsv&#39;)) |&gt;
  str_to_sentence()
astromature_humans &lt;-
  read_lines(file = here(data_dir, &#39;top_mature_astrocyte_genes_by_humans.tsv&#39;)) |&gt;
  str_to_sentence()

# aggregate:
genes.manual &lt;- unique(c(genes.embed, genes.nature,
                         genes.jj, genes.anatomy.jj,
                         astroenriched_mouse,
                         astromature_humans,
                         astroprogenitor_humans))

# ROC-test markers
srt.markers.roc &lt;- read_csv(file = here(tables_dir,
                      &#39;ageing_astr_recon-mrk_roc-sct.csv&#39;))
srt.markers.roc %&gt;%
    group_by(cluster) %&gt;%
    slice_max(n = 3, order_by = avg_log2FC)</code></pre>
<pre><code># A tibble: 9 × 8
# Groups:   cluster [3]
  myAUC avg_diff power avg_log2FC pct.1 pct.2 cluster gene         
  &lt;dbl&gt;    &lt;dbl&gt; &lt;dbl&gt;      &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;   &lt;dbl&gt; &lt;chr&gt;        
1 0.878    1.82  0.756       2.63 0.929 0.357       5 mt-Co2       
2 0.871    1.73  0.742       2.50 0.889 0.316       5 mt-Cytb      
3 0.856    1.70  0.712       2.45 0.856 0.324       5 mt-Nd1       
4 0.72     2.21  0.44        3.18 0.615 0.189       8 Gfap         
5 0.783    1.76  0.566       2.54 0.784 0.357       8 Scd2         
6 0.715    1.75  0.43        2.53 0.692 0.353       8 Aqp4         
7 0.736    1.33  0.472       1.92 0.617 0.24       10 Xrcc5        
8 0.736    1.08  0.472       1.56 0.543 0.228      10 Gm13266      
9 0.701    0.868 0.402       1.25 0.58  0.24       10 2610001J05Rik</code></pre>
<pre class="r"><code>gene_list_plot &lt;- 
  Extract_Top_Markers(marker_dataframe = srt.markers.roc,
                      num_genes = 3,
                      rank_by = &quot;avg_log2FC&quot;,
                      make_unique = TRUE,
                      named_vector = FALSE)


# LR-test markers
srt.markers.lr &lt;- read_csv(file = here(tables_dir,
                      &#39;ageing_astr_recon-mrk_logreg-sct.csv&#39;))
srt.markers.lr %&gt;%
    group_by(cluster) %&gt;%
    slice_max(n = 3, order_by = avg_log2FC)</code></pre>
<pre><code># A tibble: 23 × 7
# Groups:   cluster [9]
       p_val avg_log2FC pct.1 pct.2 p_val_adj cluster gene   
       &lt;dbl&gt;      &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;     &lt;dbl&gt;   &lt;dbl&gt; &lt;chr&gt;  
 1 1.84e- 72      1.91  0.6   0.246 6.42e- 69       1 Aldh1a1
 2 1.41e- 24      1.62  0.407 0.223 4.95e- 21       1 Esrrg  
 3 1.24e- 21      1.38  0.445 0.276 4.32e- 18       1 Kcnq3  
 4 5.25e- 19      0.920 0.476 0.294 1.84e- 15       2 Tnfaip8
 5 1.98e- 15      0.810 0.393 0.229 6.94e- 12       2 Sgcd   
 6 2.07e- 32      2.05  0.575 0.421 7.25e- 29       3 Tspan7 
 7 6.85e- 70      1.87  0.575 0.256 2.40e- 66       3 Slit2  
 8 8.72e- 51      1.86  0.514 0.164 3.05e- 47       3 Pdzph1 
 9 3.78e-  3      0.326 0.505 0.333 1   e+  0       4 Pcif1  
10 2.81e-192      2.63  0.929 0.357 9.84e-189       5 mt-Co2 
# … with 13 more rows</code></pre>
<pre class="r"><code>gene_list_plot2 &lt;- 
  Extract_Top_Markers(marker_dataframe = srt.markers.lr,
                      num_genes = 5,
                      rank_by = &quot;avg_log2FC&quot;,
                      make_unique = TRUE,
                      named_vector = FALSE)

all_integrated &lt;- LoadH5Seurat(here(raw_dir, &quot;GSE188646-reintegrated.h5Seurat&quot;))
astro &lt;- LoadH5Seurat(here(raw_dir, &quot;astro-wo-outliers.h5Seurat&quot;))</code></pre>
<pre class="r"><code>DefaultAssay(object = all_integrated) &lt;- &quot;RNA&quot;
plEmbCombWMrkR &lt;- FeaturePlot(
  all_integrated,
  features = c(&quot;Fos&quot;, &quot;Glul&quot;, &quot;Gja1&quot;, &quot;Ndrg2&quot;, 
               &quot;S100b&quot;, &quot;Tafa1&quot;, &quot;Aldh1a1&quot;, &quot;Plcb1&quot;, 
               &quot;Sgcd&quot;, &quot;Slit2&quot;, &quot;Apoe&quot;, &quot;Gfap&quot;, 
               &quot;Slc38a1&quot;, &quot;Cst3&quot;, &quot;Lxn&quot;, &quot;Mfn2&quot;),
  min.cutoff = &quot;q05&quot;, max.cutoff = &quot;q95&quot;,
  ncol = 4, order = TRUE
) + patchwork::plot_annotation(title = &quot;Astro-Marker choice (RNA)&quot;,
                               theme = theme(plot.title = element_text(size = 22)))
plEmbCombWMrkR</code></pre>
<p><img src="figure/2022-12-12_pr.Rmd/pl-all-integrated-whole-astrocytes-markers-1.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>DefaultAssay(object = all_integrated) &lt;- &quot;integrated&quot;
plEmbCombWMrkI &lt;- FeaturePlot(
  all_integrated,
  features = c(&quot;Fos&quot;, &quot;Glul&quot;, &quot;Gja1&quot;, &quot;Ndrg2&quot;, 
               &quot;S100b&quot;, &quot;Tafa1&quot;, &quot;Aldh1a1&quot;, &quot;Plcb1&quot;, 
               &quot;Sgcd&quot;, &quot;Slit2&quot;, &quot;Apoe&quot;, &quot;Gfap&quot;, 
               &quot;Slc38a1&quot;, &quot;Cst3&quot;, &quot;Lxn&quot;, &quot;Mfn2&quot;),
  min.cutoff = &quot;q05&quot;, max.cutoff = &quot;q95&quot;,
  ncol = 4, order = TRUE
) + patchwork::plot_annotation(title = &quot;Astro-Marker choice (integrated)&quot;,
                               theme = theme(plot.title = element_text(size = 22)))
plEmbCombWMrkI</code></pre>
<p><img src="figure/2022-12-12_pr.Rmd/pl-all-integrated-whole-astrocytes-markers-2.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>save_my_plot(name = &quot;ageing-pick-mrks-whole-integrated&quot;, 
             plt = plEmbCombWMrkI, 
             type = &quot;feature-plot&quot;,
             h = 12,
             asp = 1.152)</code></pre>
<pre class="r"><code>DefaultAssay(object = astro) &lt;- &quot;RNA&quot;
subset_genes &lt;- 
  astro@assays$RNA@counts |&gt;
  rowSums(na.rm = TRUE) %&gt;%
  .[. &gt; 50 | . %in% genes.manual] %&gt;% names() %&gt;%
  .[!. %in% housekeeping_mouse]
astro &lt;- DietSeurat(astro,
                    assays = &quot;RNA&quot;,
                    features = subset_genes)
# select novoseq data only to avoid integration against multiple variables
astro &lt;- 
  subset(astro, 
         subset = sample %in% c(&quot;Aged_3&quot;, &quot;Aged_4&quot;,
                                &quot;Young_3&quot;, &quot;Young_4&quot;))
astro@meta.data &lt;- 
  astro@meta.data |&gt; 
  dplyr::select(nCount_RNA, nFeature_RNA, stim, percent.mt, sample)
srt_list &lt;- Seurat::SplitObject(astro, split.by = &quot;sample&quot;)
srt_list &lt;- lapply(X = srt_list,
                   vst.flavor = &quot;v2&quot;,
                   FUN = SCTransform,
                   vars.to.regress = &quot;percent.mt&quot;,
                   method = &quot;glmGamPoi&quot;,
                   variable.features.n = 8000,
                   return.only.var.genes = FALSE,
                   seed.use = reseed,
                   verbose = FALSE)

features &lt;- 
  SelectIntegrationFeatures(object.list = srt_list, 
                            nfeatures = 4000,
                            verbose = FALSE)
all_sct_features &lt;- 
  srt_list |&gt; 
  map(pluck, &quot;assays&quot;, &quot;SCT&quot;, &quot;var.features&quot;) |&gt;
  purrr::reduce(c) |&gt; 
  unique()
keep_genes &lt;- 
  c(genes.manual,
    features) |&gt; 
  unique() %&gt;%
  .[. %in% all_sct_features]
glimpse(keep_genes)</code></pre>
<pre><code> chr [1:4040] &quot;Ndrg2&quot; &quot;Slc1a3&quot; &quot;Gfap&quot; &quot;S100b&quot; &quot;Agt&quot; &quot;Slc6a11&quot; ...</code></pre>
<pre class="r"><code>outOfAnchors &lt;- keep_genes[!keep_genes %in% features]
cat(c(&quot;\n&quot;, outOfAnchors, &quot;\n&quot;))</code></pre>
<pre><code>
 Fgfr3 Slc16a1 Slc16a7 P2rx7 Slc18a2 Slc17a5 Osmr S100a6 Rnf13 Rsph9 Rad23b Hopx Vim Slc7a10 Acsbg1 Cyp4f14 Dctd Dio2 Gpr37l1 Malat1 Nrxn1 Ntsr2 Ptprz1 Rfx4 Son Olig2 Chst2 Fam173a Rnf5 Tspan2 Tmem176a Hhatl Nudt3 Ryr3 Ppdpf Nusap1 Tnc Hes6 Lrig3 E2f5 </code></pre>
<pre class="r"><code>CheckGenesInSCT &lt;- function(gene) {
  srt_list |&gt; 
    map(pluck, &quot;assays&quot;, &quot;SCT&quot;, &quot;var.features&quot;) |&gt; 
    map(~gene %in% .x) |&gt; 
    simplify()
}
chkSamples &lt;-
  outOfAnchors |&gt;
  map(CheckGenesInSCT) |&gt;
  base::simplify2array() |&gt; 
  as_tibble()
colnames(chkSamples) &lt;- outOfAnchors
chkSamples</code></pre>
<pre><code># A tibble: 4 × 40
  Fgfr3 Slc16a1 Slc16a7 P2rx7 Slc18a2 Slc17a5 Osmr  S100a6 Rnf13 Rsph9
  &lt;lgl&gt; &lt;lgl&gt;   &lt;lgl&gt;   &lt;lgl&gt; &lt;lgl&gt;   &lt;lgl&gt;   &lt;lgl&gt; &lt;lgl&gt;  &lt;lgl&gt; &lt;lgl&gt;
1 FALSE TRUE    FALSE   TRUE  TRUE    FALSE   TRUE  TRUE   TRUE  FALSE
2 TRUE  TRUE    TRUE    TRUE  TRUE    TRUE    FALSE FALSE  FALSE FALSE
3 TRUE  TRUE    FALSE   TRUE  FALSE   TRUE    FALSE FALSE  TRUE  FALSE
4 TRUE  FALSE   FALSE   FALSE TRUE    TRUE    FALSE FALSE  TRUE  TRUE 
# … with 30 more variables: Rad23b &lt;lgl&gt;, Hopx &lt;lgl&gt;, Vim &lt;lgl&gt;,
#   Slc7a10 &lt;lgl&gt;, Acsbg1 &lt;lgl&gt;, Cyp4f14 &lt;lgl&gt;, Dctd &lt;lgl&gt;,
#   Dio2 &lt;lgl&gt;, Gpr37l1 &lt;lgl&gt;, Malat1 &lt;lgl&gt;, Nrxn1 &lt;lgl&gt;,
#   Ntsr2 &lt;lgl&gt;, Ptprz1 &lt;lgl&gt;, Rfx4 &lt;lgl&gt;, Son &lt;lgl&gt;, Olig2 &lt;lgl&gt;,
#   Chst2 &lt;lgl&gt;, Fam173a &lt;lgl&gt;, Rnf5 &lt;lgl&gt;, Tspan2 &lt;lgl&gt;,
#   Tmem176a &lt;lgl&gt;, Hhatl &lt;lgl&gt;, Nudt3 &lt;lgl&gt;, Ryr3 &lt;lgl&gt;,
#   Ppdpf &lt;lgl&gt;, Nusap1 &lt;lgl&gt;, Tnc &lt;lgl&gt;, Hes6 &lt;lgl&gt;, Lrig3 &lt;lgl&gt;, …</code></pre>
<pre class="r"><code>selected_genes &lt;-
  c(outOfAnchors[
    chkSamples |&gt;
      map(all) |&gt;
      simplify()],
    features)
glimpse(selected_genes)</code></pre>
<pre><code> chr [1:4008] &quot;Acsbg1&quot; &quot;Gpr37l1&quot; &quot;Olig2&quot; &quot;Chst2&quot; &quot;Fam173a&quot; ...</code></pre>
<pre class="r"><code>npcs &lt;- 50
srt_list &lt;-
  PrepSCTIntegration(object.list = srt_list,
                     anchor.features = selected_genes)

anchors &lt;- FindIntegrationAnchors(
  object.list = srt_list,
  normalization.method = &quot;SCT&quot;,
  anchor.features = selected_genes,
  reduction = &quot;cca&quot;,
  dims = 1:npcs,
  k.anchor = 10,
  k.score = 50,
  k.filter = 100,
  max.features = 500,
  n.trees = 100,
  verbose = FALSE
)

combined_sct &lt;- IntegrateData(
  anchorset = anchors,
  normalization.method = &quot;SCT&quot;,
  features.to.integrate = selected_genes,
  dims = 1:npcs,
  k.weight = 200,
  verbose = FALSE
)

combined_sct &lt;-
  RunPCA(combined_sct, npcs = npcs, 
         seed.use = reseed, verbose = FALSE)
combined_sct</code></pre>
<pre><code>An object of class Seurat 
26330 features across 1502 samples within 3 assays 
Active assay: integrated (4008 features, 4008 variable features)
 2 other assays present: RNA, SCT
 1 dimensional reduction calculated: pca</code></pre>
<pre class="r"><code>combined_sct &lt;- combined_sct |&gt; 
  RunUMAP(reduction = &quot;pca&quot;, return.model = TRUE,
          densmap = TRUE, dims = 1:npcs, 
          seed.use = reseed, verbose = FALSE) |&gt; 
  FindNeighbors(reduction = &quot;pca&quot;, dims = 1:npcs, verbose = FALSE)</code></pre>
<p>Plot by source</p>
<pre class="r"><code>plEmbCombBatch &lt;- DimPlot(combined_sct, reduction = &quot;umap&quot;,
                          group.by = &quot;sample&quot;,
                          label = TRUE, repel = TRUE) + NoLegend()
plEmbCombReg &lt;- DimPlot(combined_sct, reduction = &quot;umap&quot;,
                        group.by = &quot;stim&quot;,
                        label = TRUE, repel = TRUE) + NoLegend()
plEmbCombBatch + plEmbCombReg</code></pre>
<p><img src="figure/2022-12-12_pr.Rmd/pl-all-integrated-astrocytes-sample-1.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code># plot &lt;- DimPlot(combined_sct, reduction = &quot;umap&quot;)
# select.cells &lt;- CellSelector(plot = plot)
select.cells &lt;-
  WhichCells(
    combined_sct, 
    expression = UMAP_1 &gt; 5)
Idents(combined_sct, cells = select.cells) &lt;- &quot;deleteme&quot;</code></pre>
<pre class="r"><code># Now, we find markers that are specific to the selected cells, and find clear duplets markers
combined_sct &lt;-
  PrepSCTFindMarkers(combined_sct, assay = &quot;SCT&quot;)
newcells.markers &lt;-
  FindMarkers(
    combined_sct, 
    assay = &quot;SCT&quot;,
    ident.1 = &quot;deleteme&quot;,
    ident.2 = NULL,
    min.diff.pct = 0.3,
    only.pos = TRUE
  )
head(newcells.markers)</code></pre>
<pre><code>               p_val avg_log2FC pct.1 pct.2     p_val_adj
Dlgap2 2.068111e-124  1.1246489 0.600 0.015 2.307391e-120
Nrg3os 3.267109e-122  0.8564753 0.538 0.010 3.645113e-118
Dscam  3.713380e-115  1.6732427 0.815 0.049 4.143018e-111
Ksr2   4.170068e-106  0.8805369 0.646 0.026 4.652545e-102
Kcnb2  1.994777e-104  1.6591216 0.754 0.046 2.225573e-100
Xkr4   2.888252e-103  1.6450738 0.862 0.069  3.222423e-99</code></pre>
<pre class="r"><code>FeaturePlot(
  combined_sct,
  features = c(&quot;Fos&quot;, &quot;Glul&quot;, &quot;Gja1&quot;, &quot;Ndrg2&quot;, 
               &quot;S100b&quot;, &quot;Tafa1&quot;, &quot;Aldh1a1&quot;, &quot;Plcb1&quot;, 
               &quot;Sgcd&quot;, &quot;Slit2&quot;, &quot;Apoe&quot;, &quot;Gfap&quot;, 
               &quot;Slc38a1&quot;, &quot;Cst3&quot;, &quot;Lxn&quot;, &quot;Mfn2&quot;),
  min.cutoff = &quot;q05&quot;, max.cutoff = &quot;q95&quot;,
  ncol = 4, order = TRUE
) + 
  patchwork::plot_annotation(
    title = &quot;Astro-Marker choice (integrated astrocytes from novoseq)&quot;,
    theme = theme(plot.title = element_text(size = 22))
    )</code></pre>
<p><img src="figure/2022-12-12_pr.Rmd/pl-astro-integrated-contam-astrocytes-markers-1.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>combined_sct2 &lt;- subset(combined_sct, idents = c(&quot;deleteme&quot;), invert = TRUE)
DimPlot(combined_sct2, label = T, repel = T, group.by = &quot;sample&quot;)</code></pre>
<p><img src="figure/2022-12-12_pr.Rmd/delete-outliers-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>combined_sct &lt;- combined_sct2
rm(combined_sct2)</code></pre>
<pre class="r"><code>srt_list &lt;- 
  Seurat::SplitObject(combined_sct, split.by = &quot;sample&quot;)
srt_list &lt;- lapply(X = srt_list,
                   vst.flavor = &quot;v2&quot;,
                   FUN = SCTransform,
                   vars.to.regress = &quot;percent.mt&quot;,
                   method = &quot;glmGamPoi&quot;,
                   variable.features.n = 8000,
                   return.only.var.genes = FALSE,
                   seed.use = reseed,
                   verbose = FALSE)

features &lt;- 
  SelectIntegrationFeatures(object.list = srt_list, 
                            nfeatures = 4000,
                            verbose = FALSE)
all_sct_features &lt;- 
  srt_list |&gt; 
  map(pluck, &quot;assays&quot;, &quot;SCT&quot;, &quot;var.features&quot;) |&gt;
  purrr::reduce(c) |&gt; 
  unique()
keep_genes &lt;- 
  c(genes.manual,
    features) |&gt; 
  unique() %&gt;%
  .[. %in% all_sct_features]
glimpse(keep_genes)</code></pre>
<pre><code> chr [1:4059] &quot;Ndrg2&quot; &quot;Slc1a3&quot; &quot;Gfap&quot; &quot;S100b&quot; &quot;Agt&quot; &quot;Slc6a11&quot; ...</code></pre>
<pre class="r"><code>outOfAnchors &lt;- keep_genes[!keep_genes %in% features]
cat(c(&quot;\n&quot;, outOfAnchors, &quot;\n&quot;))</code></pre>
<pre><code>
 Ndrg2 Slc6a11 Fgfr3 Slc16a1 Slc16a7 P2rx7 Slc18a2 Slc17a5 S100a6 Rdh10 Rnf13 Rsph9 Hopx 2810459M11Rik Vim Slc7a10 Acsbg1 Acsl3 Cbs Ckb Cyp4f14 Dbx2 Dctd Dio2 Gjb6 Gm266 Gpr37l1 Grhl1 Hapln1 Malat1 Mfge8 Nrxn1 Ntsr2 Otx2 Prodh Ptprz1 Rfx4 Son Sox2 Snhg11 Lxn Olig2 Thrsp Vcam1 Chst2 Magoh Fam173a Rnf5 Tspan2 Tmem176a Hhatl Nudt3 Fbxo2 Ryr3 Ppdpf Nusap1 Tnc Hes6 E2f5 </code></pre>
<pre class="r"><code>chkSamples &lt;-
  outOfAnchors |&gt;
  map(CheckGenesInSCT) |&gt;
  base::simplify2array() |&gt; 
  as_tibble()
colnames(chkSamples) &lt;- outOfAnchors
chkSamples</code></pre>
<pre><code># A tibble: 4 × 59
  Ndrg2 Slc6a11 Fgfr3 Slc16a1 Slc16a7 P2rx7 Slc18a2 Slc17a5 S100a6
  &lt;lgl&gt; &lt;lgl&gt;   &lt;lgl&gt; &lt;lgl&gt;   &lt;lgl&gt;   &lt;lgl&gt; &lt;lgl&gt;   &lt;lgl&gt;   &lt;lgl&gt; 
1 TRUE  FALSE   FALSE TRUE    FALSE   TRUE  TRUE    FALSE   TRUE  
2 FALSE TRUE    FALSE TRUE    TRUE    TRUE  TRUE    TRUE    FALSE 
3 TRUE  TRUE    TRUE  TRUE    FALSE   TRUE  FALSE   TRUE    FALSE 
4 TRUE  TRUE    TRUE  FALSE   FALSE   FALSE TRUE    TRUE    FALSE 
# … with 50 more variables: Rdh10 &lt;lgl&gt;, Rnf13 &lt;lgl&gt;, Rsph9 &lt;lgl&gt;,
#   Hopx &lt;lgl&gt;, `2810459M11Rik` &lt;lgl&gt;, Vim &lt;lgl&gt;, Slc7a10 &lt;lgl&gt;,
#   Acsbg1 &lt;lgl&gt;, Acsl3 &lt;lgl&gt;, Cbs &lt;lgl&gt;, Ckb &lt;lgl&gt;, Cyp4f14 &lt;lgl&gt;,
#   Dbx2 &lt;lgl&gt;, Dctd &lt;lgl&gt;, Dio2 &lt;lgl&gt;, Gjb6 &lt;lgl&gt;, Gm266 &lt;lgl&gt;,
#   Gpr37l1 &lt;lgl&gt;, Grhl1 &lt;lgl&gt;, Hapln1 &lt;lgl&gt;, Malat1 &lt;lgl&gt;,
#   Mfge8 &lt;lgl&gt;, Nrxn1 &lt;lgl&gt;, Ntsr2 &lt;lgl&gt;, Otx2 &lt;lgl&gt;, Prodh &lt;lgl&gt;,
#   Ptprz1 &lt;lgl&gt;, Rfx4 &lt;lgl&gt;, Son &lt;lgl&gt;, Sox2 &lt;lgl&gt;, Snhg11 &lt;lgl&gt;, …</code></pre>
<pre class="r"><code>selected_genes &lt;-
  c(outOfAnchors[
    chkSamples |&gt;
      map(all) |&gt;
      simplify()],
    features)
glimpse(selected_genes)</code></pre>
<pre><code> chr [1:4012] &quot;Acsl3&quot; &quot;Gjb6&quot; &quot;Gm266&quot; &quot;Gpr37l1&quot; &quot;Hapln1&quot; &quot;Sox2&quot; ...</code></pre>
<pre class="r"><code>npcs &lt;- 50
srt_list &lt;-
  PrepSCTIntegration(object.list = srt_list,
                     anchor.features = selected_genes)

anchors &lt;- FindIntegrationAnchors(
  object.list = srt_list,
  normalization.method = &quot;SCT&quot;,
  anchor.features = selected_genes,
  reduction = &quot;cca&quot;,
  dims = 1:npcs,
  k.anchor = 10,
  k.score = 50,
  k.filter = 100,
  max.features = 500,
  n.trees = 100,
  verbose = FALSE
)

combined_sct &lt;- IntegrateData(
  anchorset = anchors,
  normalization.method = &quot;SCT&quot;,
  features.to.integrate = selected_genes,
  dims = 1:npcs,
  k.weight = 200,
  verbose = FALSE
)</code></pre>
<pre class="r"><code>npcs &lt;- 50
combined_sct &lt;- combined_sct |&gt; 
  RunPCA(seed.use = reseed, verbose = FALSE) |&gt; 
  RunUMAP(reduction = &quot;pca&quot;, return.model = TRUE,
          densmap = TRUE, dims = 1:npcs, 
          seed.use = reseed, verbose = FALSE) |&gt; 
  FindNeighbors(reduction = &quot;pca&quot;, dims = 1:npcs, verbose = FALSE)</code></pre>
<p>Plot by source after clean up</p>
<pre class="r"><code>plEmbCombBatch &lt;- DimPlot(combined_sct, reduction = &quot;umap&quot;,
                          group.by = &quot;sample&quot;,
                          label = TRUE, repel = TRUE) + NoLegend()
plEmbCombReg &lt;- DimPlot(combined_sct, reduction = &quot;umap&quot;,
                        group.by = &quot;stim&quot;,
                        label = TRUE, repel = TRUE) + NoLegend()
plEmbCombBatch + plEmbCombReg</code></pre>
<p><img src="figure/2022-12-12_pr.Rmd/pl-all-integrated-astrocytes-sample-clean-1.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>metadata &lt;- combined_sct@meta.data
rownames(metadata) &lt;- colnames(combined_sct)
ref.labels &lt;- metadata$stim
library(mrtree)
resolutions &lt;-
  modularity_event_sampling(
    A = combined_sct@graphs$integrated_snn,
    n.res = 15,
    gamma.min = 0.1,
    gamma.max = 2
  ) # sample based on the similarity matrix

# clustering using Suerat
combined_sct &lt;- combined_sct |&gt; 
  FindClusters(algorithm = 4, method = &quot;igraph&quot;,
               resolution = resolutions, random.seed = reseed,
               verbose = FALSE)

# initial cluster tree from Seurat flat clustering
plot_clustree(
  labelmat = combined_sct@meta.data,
  prefix = &#39;integrated_snn_res.&#39;,
  ref.labels = ref.labels,
  plot.ref = FALSE
)</code></pre>
<p><img src="figure/2022-12-12_pr.Rmd/pl-clustree-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Adjusted Multiresolution Rand Index (AMRI)
ks.flat &lt;-  apply(
  out$labelmat.flat,
  2,
  FUN = function(x)
    length(unique(x))
)
ks.mrtree &lt;-  apply(
  out$labelmat.mrtree,
  2,
  FUN = function(x)
    length(unique(x))
)
amri.flat &lt;-  sapply(1:ncol(out$labelmat.flat), function(i)
  AMRI(out$labelmat.flat[, i], ref.labels)$amri)
amri.flat &lt;-  aggregate(amri.flat, by = list(k = ks.flat), FUN = mean)
amri.recon &lt;-  sapply(1:ncol(out$labelmat.mrtree), function(i)
  AMRI(out$labelmat.mrtree[, i], ref.labels)$amri)

df &lt;-  rbind(
  data.frame(
    k = amri.flat$k,
    amri = amri.flat$x,
    method = &#39;Seurat flat&#39;
  ),
  data.frame(k = ks.mrtree, amri = amri.recon, method = &#39;MRtree&#39;)
)
ggplot2::ggplot(data = df, aes(x = k, y = amri, color = method)) + geom_line() + theme_bw()</code></pre>
<p><img src="figure/2022-12-12_pr.Rmd/pl-clustering-amri-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>stab.out &lt;- stability_plot(out)
stab.out$plot</code></pre>
<p><img src="figure/2022-12-12_pr.Rmd/pl-clustering-resolution-1.png" width="480" style="display: block; margin: auto;" /></p>
<pre class="r"><code>stab.out$df</code></pre>
<pre><code>  resolution       ari
1          1 1.0000000
2          2 0.7523158
3          3 0.7107602
4          5 0.8777208
5          6 0.8648309
6          8 0.7898090
7         13 0.7361102
8         21 0.7480721</code></pre>
<pre class="r"><code>resK &lt;-
  stab.out$df |&gt;
  as_tibble() |&gt;
  filter(ari != 1) |&gt; 
  top_n(n = 1, wt = ari) |&gt; 
  purrr::pluck(1)
resK</code></pre>
<pre><code>[1] 5</code></pre>
<pre class="r"><code>table(as.factor(out$labelmat.mrtree[, sprintf(&quot;K%s&quot;,resK)]))</code></pre>
<pre><code>
  1   2   3   4   5 
421 449 281 154 132 </code></pre>
<pre class="r"><code>combined_sct$k_tree &lt;- as.factor(out$labelmat.mrtree[, sprintf(&quot;K%s&quot;,resK)])
p1 &lt;- DimPlot(combined_sct, label = T, repel = T) + ggtitle(&quot;Unsupervised clustering&quot;) + NoLegend()
p2 &lt;- DimPlot(combined_sct, label = T, repel = T, group.by = &quot;k_tree&quot;) + ggtitle(&quot;MRTree&quot;) + NoLegend()

p1 | p2</code></pre>
<p><img src="figure/2022-12-12_pr.Rmd/pl-clustering-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>Idents(combined_sct) &lt;- &quot;k_tree&quot;
combined_sct &lt;- 
  PrepSCTFindMarkers(combined_sct, assay = &quot;SCT&quot;)

srt.markers.roc &lt;- 
  FindAllMarkers(
    combined_sct,
    assay = &quot;SCT&quot;,
    only.pos = TRUE,
    min.diff.pct = 0.1,
    logfc.threshold = 0.25,
    test.use = &quot;roc&quot;)
write_csv(srt.markers.roc,
          here(tables_dir,
               &#39;novoseq-ageing_astr_recon-fix-mrk_roc-sct.csv&#39;))

srt.markers.roc %&gt;%
    group_by(cluster) %&gt;%
    slice_max(n = 3, order_by = avg_log2FC)</code></pre>
<pre><code># A tibble: 2 × 8
# Groups:   cluster [2]
  myAUC avg_diff power avg_log2FC pct.1 pct.2 cluster gene   
  &lt;dbl&gt;    &lt;dbl&gt; &lt;dbl&gt;      &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;fct&gt;   &lt;chr&gt;  
1 0.738    0.601 0.476      0.867 0.651 0.206 3       Aldh1a1
2 0.727    0.578 0.454      0.833 0.805 0.504 4       Gpc6   </code></pre>
<pre class="r"><code>srt.markers.lr &lt;-
  FindAllMarkers(
    combined_sct,
    assay = &quot;SCT&quot;,
    only.pos = TRUE,
    min.diff.pct = 0.15,
    logfc.threshold = 0.25,
    test.use = &quot;LR&quot;,
    latent.vars = &quot;sample&quot;)
readr::write_csv(srt.markers.lr, 
                 here(tables_dir,
                      &#39;novoseq-ageing_astr_recon-fix-mrk_logreg-sct.csv&#39;))
srt.markers.lr %&gt;%
    group_by(cluster) %&gt;%
    slice_max(n = 3, order_by = avg_log2FC)</code></pre>
<pre><code># A tibble: 13 × 7
# Groups:   cluster [5]
      p_val avg_log2FC pct.1 pct.2 p_val_adj cluster gene   
      &lt;dbl&gt;      &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;     &lt;dbl&gt; &lt;fct&gt;   &lt;chr&gt;  
 1 4.42e-20      0.484 0.798 0.639  4.91e-16 1       Lars2  
 2 3.50e-19      0.375 0.586 0.338  3.89e-15 2       Gm29683
 3 2.86e-10      0.325 0.332 0.173  3.17e- 6 2       Sgcd   
 4 1.53e-12      0.304 0.318 0.153  1.69e- 8 2       Tshz2  
 5 6.11e-49      0.867 0.651 0.206  6.79e-45 3       Aldh1a1
 6 2.95e-25      0.530 0.349 0.105  3.28e-21 3       Tafa1  
 7 1.68e-19      0.486 0.299 0.099  1.87e-15 3       Sntg1  
 8 5.14e-30      0.918 0.455 0.108  5.71e-26 4       Slc38a1
 9 1.73e-24      0.906 0.571 0.204  1.92e-20 4       Slit2  
10 1.13e-22      0.833 0.805 0.504  1.26e-18 4       Gpc6   
11 6.77e- 7      0.369 0.841 0.555  7.52e- 3 5       Agt    
12 3.65e- 4      0.307 0.659 0.465  1   e+ 0 5       Calm1  
13 5.37e- 5      0.300 0.47  0.267  5.97e- 1 5       Tenm4  </code></pre>
<pre class="r"><code># srt.markers.mast &lt;-
#   FindAllMarkers(
#     combined_sct,
#     assay = &quot;SCT&quot;,
#     only.pos = TRUE,
#     min.diff.pct = 0.15,
#     logfc.threshold = 0.25,
#     test.use = &quot;MAST&quot;,
#     latent.vars = &quot;sample&quot;)
# readr::write_csv(srt.markers.mast, 
#                  here(tables_dir,
#                       &#39;novoseq-ageing_astr_recon-fix-mrk_mast-sct.csv&#39;))
# srt.markers.mast %&gt;%
#     group_by(cluster) %&gt;%
#     slice_max(n = 3, order_by = avg_log2FC)</code></pre>
<pre class="r"><code>FeaturePlot(
  combined_sct,
  features = c(&quot;Apoe&quot;, &quot;Gfap&quot;, &quot;Slit2&quot;, &quot;Aldh1a1&quot;, 
               &quot;Tafa1&quot;, &quot;Plcb1&quot;, &quot;Sgcd&quot;, &quot;Slc38a1&quot;, 
               &quot;Fos&quot;, &quot;Gja1&quot;, &quot;Snap25&quot;, &quot;Olig1&quot;, 
               &quot;Per2&quot;, &quot;Lars2&quot;, &quot;Lxn&quot;, &quot;Mfn2&quot;),
  min.cutoff = &quot;q05&quot;, max.cutoff = &quot;q95&quot;,
  ncol = 4, order = TRUE
) + 
  patchwork::plot_annotation(
    title = &quot;Astro-Marker choice (integrated astrocytes from novoseq)&quot;,
    theme = theme(plot.title = element_text(size = 22))
    )</code></pre>
<p><img src="figure/2022-12-12_pr.Rmd/pl-astro-integrated-filt-astrocytes-markers-1.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>hypoth.nuclei.astro &lt;- LoadH5Seurat(file = here(data_dir, &quot;comb_astrocytes.bak.h5Seurat&quot;))
DefaultAssay(hypoth.nuclei.astro) &lt;- &quot;SCT&quot;
hypoth.nuclei.astro &lt;-
  hypoth.nuclei.astro |&gt; 
  DietSeurat(assays = c(&quot;RNA&quot;, &quot;SCT&quot;))
hypoth.list &lt;-
  Seurat::SplitObject(
    hypoth.nuclei.astro,
    split.by = &quot;Batch_ID&quot;)</code></pre>
<div id="arcuate-nucleus" class="section level2">
<h2>Arcuate nucleus</h2>
<div id="campbell-et-al.-2017-dropseq" class="section level3">
<h3>Campbell et al. 2017 Dropseq</h3>
<pre class="r"><code>hypoth.query.campbell_2017.1 &lt;-
  hypoth.list[[&quot;CampbellDropseq_batch_1&quot;]]
hypoth.anchors.campbell_2017.1 &lt;-
  FindTransferAnchors(
    reference = combined_sct,
    query = hypoth.query.campbell_2017.1,
    normalization.method = &quot;SCT&quot;,
    dims = 1:npcs,
    k.anchor = 10,
    k.score = 50,
    k.filter = 100,
    max.features = 500,
    n.trees = 100,
    reference.reduction = &quot;pca&quot;)

predictions.campbell_2017.1 &lt;-
  TransferData(
    anchorset = hypoth.anchors.campbell_2017.1,
    refdata = combined_sct$k_tree,
    dims = 1:npcs)

hypoth.query.campbell_2017.1 &lt;-
  MapQuery(
    anchorset = hypoth.anchors.campbell_2017.1,
    reference = combined_sct,
    query = hypoth.query.campbell_2017.1,
    refdata = list(k_tree = &quot;k_tree&quot;),
    reference.reduction = &quot;pca&quot;,
    reduction.model = &quot;umap&quot;)</code></pre>
<pre class="r"><code>p1 &lt;- DimPlot(combined_sct, reduction = &quot;umap&quot;, group.by = &quot;k_tree&quot;, label = TRUE, label.size = 3,
    repel = TRUE) + NoLegend() + ggtitle(&quot;Reference annotations&quot;)
p.campbell_2017.1 &lt;- DimPlot(hypoth.query.campbell_2017.1, reduction = &quot;ref.umap&quot;, group.by = &quot;predicted.k_tree&quot;, label = TRUE,
    label.size = 3, repel = TRUE) + NoLegend() + ggtitle(&quot;Query transferred labels&quot;)
p1 + p.campbell_2017.1</code></pre>
<p><img src="figure/2022-12-12_pr.Rmd/pl-transfer-ageing-campbell-2017-1-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="lutomska-et-al.-2017-dropseq" class="section level3">
<h3>Lutomska et al. 2017 Dropseq</h3>
<pre class="r"><code>hypoth.query.lutomska_2022.1 &lt;-
  hypoth.list[[&quot;Lutomska_chow_10xv3&quot;]]
hypoth.anchors.lutomska_2022.1 &lt;-
  FindTransferAnchors(
    reference = combined_sct,
    query = hypoth.query.lutomska_2022.1,
    normalization.method = &quot;SCT&quot;,
    dims = 1:npcs,
    k.anchor = 10,
    k.score = 50,
    k.filter = 100,
    max.features = 500,
    n.trees = 100,
    reference.reduction = &quot;pca&quot;)

predictions.lutomska_2022.1 &lt;-
  TransferData(
    anchorset = hypoth.anchors.lutomska_2022.1,
    refdata = combined_sct$k_tree,
    dims = 1:npcs)

hypoth.query.lutomska_2022.1 &lt;-
  MapQuery(
    anchorset = hypoth.anchors.lutomska_2022.1,
    reference = combined_sct,
    query = hypoth.query.lutomska_2022.1,
    refdata = list(k_tree = &quot;k_tree&quot;),
    reference.reduction = &quot;pca&quot;,
    reduction.model = &quot;umap&quot;)</code></pre>
<pre class="r"><code>p.lutomska_2022.1 &lt;-
  DimPlot(
    hypoth.query.lutomska_2022.1,
    reduction = &quot;ref.umap&quot;,
    group.by = &quot;predicted.k_tree&quot;,
    label = TRUE,
    label.size = 3,
    repel = TRUE) +
  NoLegend() +
  ggtitle(&quot;Query transferred labels&quot;)

p1 + p.lutomska_2022.1</code></pre>
<p><img src="figure/2022-12-12_pr.Rmd/pl-transfer-ageing-lutomska-2022-1-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="deng-et-al.-2021-10xv3" class="section level3">
<h3>Deng et al. 2021 10xV3</h3>
<pre class="r"><code>hypoth.query.deng_2021.1 &lt;-
  hypoth.list[[&quot;Deng10xv3_536-1_chow-diet&quot;]]
hypoth.anchors.deng_2021.1 &lt;-
  FindTransferAnchors(
    reference = combined_sct,
    query = hypoth.query.deng_2021.1,
    normalization.method = &quot;SCT&quot;,
    dims = 1:npcs,
    k.anchor = 10,
    k.score = 50,
    k.filter = 100,
    max.features = 500,
    n.trees = 100,
    reference.reduction = &quot;pca&quot;)

predictions.deng_2021.1 &lt;-
  TransferData(
    anchorset = hypoth.anchors.deng_2021.1,
    refdata = combined_sct$k_tree,
    dims = 1:npcs)

hypoth.query.deng_2021.1 &lt;-
  MapQuery(
    anchorset = hypoth.anchors.deng_2021.1,
    reference = combined_sct,
    query = hypoth.query.deng_2021.1,
    refdata = list(k_tree = &quot;k_tree&quot;),
    reference.reduction = &quot;pca&quot;,
    reduction.model = &quot;umap&quot;)</code></pre>
<pre class="r"><code>p.deng_2021.1 &lt;-
  DimPlot(
    hypoth.query.deng_2021.1,
    reduction = &quot;ref.umap&quot;,
    group.by = &quot;predicted.k_tree&quot;,
    label = TRUE,
    label.size = 3,
    repel = TRUE) +
  NoLegend() +
  ggtitle(&quot;Query transferred labels&quot;)

p1 + p.deng_2021.1</code></pre>
<p><img src="figure/2022-12-12_pr.Rmd/pl-transfer-ageing-deng-2021-1-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>hypoth.query.deng_2021.3 &lt;-
  hypoth.list[[&quot;Deng10xv3_536-3_chow-diet&quot;]]
hypoth.anchors.deng_2021.3 &lt;-
  FindTransferAnchors(
    reference = combined_sct,
    query = hypoth.query.deng_2021.3,
    normalization.method = &quot;SCT&quot;,
    dims = 1:npcs,
    k.anchor = 10,
    k.score = 50,
    k.filter = 100,
    max.features = 500,
    n.trees = 100,
    reference.reduction = &quot;pca&quot;)

predictions.deng_2021.3 &lt;-
  TransferData(
    anchorset = hypoth.anchors.deng_2021.3,
    refdata = combined_sct$k_tree,
    dims = 1:npcs)

hypoth.query.deng_2021.3 &lt;-
  MapQuery(
    anchorset = hypoth.anchors.deng_2021.3,
    reference = combined_sct,
    query = hypoth.query.deng_2021.3,
    refdata = list(k_tree = &quot;k_tree&quot;),
    reference.reduction = &quot;pca&quot;,
    reduction.model = &quot;umap&quot;)</code></pre>
<pre class="r"><code>p.deng_2021.3 &lt;-
  DimPlot(
    hypoth.query.deng_2021.3,
    reduction = &quot;ref.umap&quot;,
    group.by = &quot;predicted.k_tree&quot;,
    label = TRUE,
    label.size = 3,
    repel = TRUE) +
  NoLegend() +
  ggtitle(&quot;Query transferred labels&quot;)

p1 + p.deng_2021.3</code></pre>
<p><img src="figure/2022-12-12_pr.Rmd/pl-transfer-ageing-deng-2021-3-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>hypoth.query.deng_2021.5 &lt;-
  hypoth.list[[&quot;Deng10xv3_536-5_chow-diet&quot;]]
hypoth.anchors.deng_2021.5 &lt;-
  FindTransferAnchors(
    reference = combined_sct,
    query = hypoth.query.deng_2021.5,
    normalization.method = &quot;SCT&quot;,
    dims = 1:npcs,
    k.anchor = 10,
    k.score = 50,
    k.filter = 100,
    max.features = 500,
    n.trees = 100,
    reference.reduction = &quot;pca&quot;)

predictions.deng_2021.5 &lt;-
  TransferData(
    anchorset = hypoth.anchors.deng_2021.5,
    refdata = combined_sct$k_tree,
    dims = 1:npcs)

hypoth.query.deng_2021.5 &lt;-
  MapQuery(
    anchorset = hypoth.anchors.deng_2021.5,
    reference = combined_sct,
    query = hypoth.query.deng_2021.5,
    refdata = list(k_tree = &quot;k_tree&quot;),
    reference.reduction = &quot;pca&quot;,
    reduction.model = &quot;umap&quot;)</code></pre>
<pre class="r"><code>p.deng_2021.5 &lt;-
  DimPlot(
    hypoth.query.deng_2021.5,
    reduction = &quot;ref.umap&quot;,
    group.by = &quot;predicted.k_tree&quot;,
    label = TRUE,
    label.size = 3,
    repel = TRUE) +
  NoLegend() +
  ggtitle(&quot;Query transferred labels&quot;)

p1 + p.deng_2021.5</code></pre>
<p><img src="figure/2022-12-12_pr.Rmd/pl-transfer-ageing-deng-2021-5-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="mediobasal-hypothalamus" class="section level2">
<h2>Mediobasal hypothalamus</h2>
<div id="rupp-et-al.-2021-10xv3" class="section level3">
<h3>Rupp et al. 2021 10xV3</h3>
<pre class="r"><code>hypoth.query.rupp_2021.1 &lt;-
  hypoth.list[[&quot;Rupp10x_batch_1&quot;]]
hypoth.anchors.rupp_2021.1 &lt;-
  FindTransferAnchors(
    reference = combined_sct,
    query = hypoth.query.rupp_2021.1,
    normalization.method = &quot;SCT&quot;,
    dims = 1:npcs,
    k.anchor = 10,
    k.score = 50,
    k.filter = 100,
    max.features = 500,
    n.trees = 100,
    reference.reduction = &quot;pca&quot;)

predictions.rupp_2021.1 &lt;-
  TransferData(
    anchorset = hypoth.anchors.rupp_2021.1,
    refdata = combined_sct$k_tree,
    dims = 1:npcs)

hypoth.query.rupp_2021.1 &lt;-
  MapQuery(
    anchorset = hypoth.anchors.rupp_2021.1,
    reference = combined_sct,
    query = hypoth.query.rupp_2021.1,
    refdata = list(k_tree = &quot;k_tree&quot;),
    reference.reduction = &quot;pca&quot;,
    reduction.model = &quot;umap&quot;)</code></pre>
<pre class="r"><code>p.rupp_2021.1 &lt;-
  DimPlot(
    hypoth.query.rupp_2021.1,
    reduction = &quot;ref.umap&quot;,
    group.by = &quot;predicted.k_tree&quot;,
    label = TRUE,
    label.size = 3,
    repel = TRUE) +
  NoLegend() +
  ggtitle(&quot;Query transferred labels&quot;)

p1 + p.rupp_2021.1</code></pre>
<p><img src="figure/2022-12-12_pr.Rmd/pl-transfer-ageing-rupp-2021-1-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="ventromedial-hypothalamus" class="section level2">
<h2>Ventromedial hypothalamus</h2>
<div id="affinati-et-al.-2021-10xv3-nuclei" class="section level3">
<h3>Affinati et al. 2021 10xV3 nuclei</h3>
<pre class="r"><code>hypoth.query.affinati_2021.1 &lt;-
  hypoth.list[[&quot;Affinati10x_batch_1&quot;]]
hypoth.anchors.affinati_2021.1 &lt;-
  FindTransferAnchors(
    reference = combined_sct,
    query = hypoth.query.affinati_2021.1,
    normalization.method = &quot;SCT&quot;,
    dims = 1:npcs,
    k.anchor = 10,
    k.score = 50,
    k.filter = 100,
    max.features = 500,
    n.trees = 100,
    reference.reduction = &quot;pca&quot;)

predictions.affinati_2021.1 &lt;-
  TransferData(
    anchorset = hypoth.anchors.affinati_2021.1,
    refdata = combined_sct$k_tree,
    dims = 1:npcs)

hypoth.query.affinati_2021.1 &lt;-
  MapQuery(
    anchorset = hypoth.anchors.affinati_2021.1,
    reference = combined_sct,
    query = hypoth.query.affinati_2021.1,
    refdata = list(k_tree = &quot;k_tree&quot;),
    reference.reduction = &quot;pca&quot;,
    reduction.model = &quot;umap&quot;)</code></pre>
<pre class="r"><code>p.affinati_2021.1 &lt;-
  DimPlot(
    hypoth.query.affinati_2021.1,
    reduction = &quot;ref.umap&quot;,
    group.by = &quot;predicted.k_tree&quot;,
    label = TRUE,
    label.size = 3,
    repel = TRUE) +
  NoLegend() +
  ggtitle(&quot;Query transferred labels&quot;)

p1 + p.affinati_2021.1</code></pre>
<p><img src="figure/2022-12-12_pr.Rmd/pl-transfer-ageing-affinati-2021-1-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>hypoth.query.affinati_2021.3 &lt;-
  hypoth.list[[&quot;Affinati10x_batch_3&quot;]]
hypoth.anchors.affinati_2021.3 &lt;-
  FindTransferAnchors(
    reference = combined_sct,
    query = hypoth.query.affinati_2021.3,
    normalization.method = &quot;SCT&quot;,
    dims = 1:npcs,
    k.anchor = 10,
    k.score = 50,
    k.filter = 100,
    max.features = 500,
    n.trees = 100,
    reference.reduction = &quot;pca&quot;)

predictions.affinati_2021.3 &lt;-
  TransferData(
    anchorset = hypoth.anchors.affinati_2021.3,
    refdata = combined_sct$k_tree,
    dims = 1:npcs)

hypoth.query.affinati_2021.3 &lt;-
  MapQuery(
    anchorset = hypoth.anchors.affinati_2021.3,
    reference = combined_sct,
    query = hypoth.query.affinati_2021.3,
    refdata = list(k_tree = &quot;k_tree&quot;),
    reference.reduction = &quot;pca&quot;,
    reduction.model = &quot;umap&quot;)</code></pre>
<pre class="r"><code>p.affinati_2021.3 &lt;-
  DimPlot(
    hypoth.query.affinati_2021.3,
    reduction = &quot;ref.umap&quot;,
    group.by = &quot;predicted.k_tree&quot;,
    label = TRUE,
    label.size = 3,
    repel = TRUE) +
  NoLegend() +
  ggtitle(&quot;Query transferred labels&quot;)

p1 + p.affinati_2021.3</code></pre>
<p><img src="figure/2022-12-12_pr.Rmd/pl-transfer-ageing-affinati-2021-3-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>hypoth.query.affinati_2021.4 &lt;-
  hypoth.list[[&quot;Affinati10x_batch_4&quot;]]
hypoth.anchors.affinati_2021.4 &lt;-
  FindTransferAnchors(
    reference = combined_sct,
    query = hypoth.query.affinati_2021.4,
    normalization.method = &quot;SCT&quot;,
    dims = 1:npcs,
    k.anchor = 10,
    k.score = 50,
    k.filter = 100,
    max.features = 500,
    n.trees = 100,
    reference.reduction = &quot;pca&quot;)

predictions.affinati_2021.4 &lt;-
  TransferData(
    anchorset = hypoth.anchors.affinati_2021.4,
    refdata = combined_sct$k_tree,
    dims = 1:npcs)

hypoth.query.affinati_2021.4 &lt;-
  MapQuery(
    anchorset = hypoth.anchors.affinati_2021.4,
    reference = combined_sct,
    query = hypoth.query.affinati_2021.4,
    refdata = list(k_tree = &quot;k_tree&quot;),
    reference.reduction = &quot;pca&quot;,
    reduction.model = &quot;umap&quot;)</code></pre>
<pre class="r"><code>p.affinati_2021.4 &lt;-
  DimPlot(
    hypoth.query.affinati_2021.4,
    reduction = &quot;ref.umap&quot;,
    group.by = &quot;predicted.k_tree&quot;,
    label = TRUE,
    label.size = 3,
    repel = TRUE) +
  NoLegend() +
  ggtitle(&quot;Query transferred labels&quot;)

p1 + p.affinati_2021.4</code></pre>
<p><img src="figure/2022-12-12_pr.Rmd/pl-transfer-ageing-affinati-2021-4-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="kim-et-al.-2019-10xv2" class="section level3">
<h3>Kim et al. 2019 10xV2</h3>
<pre class="r"><code>hypoth.query.kim_2019.1 &lt;-
  hypoth.list[[&quot;Kim10x_batch_1&quot;]]
hypoth.anchors.kim_2019.1 &lt;-
  FindTransferAnchors(
    reference = combined_sct,
    query = hypoth.query.kim_2019.1,
    normalization.method = &quot;SCT&quot;,
    dims = 1:npcs,
    k.anchor = 10,
    k.score = 50,
    k.filter = 100,
    max.features = 500,
    n.trees = 100,
    reference.reduction = &quot;pca&quot;)

predictions.kim_2019.1 &lt;-
  TransferData(
    anchorset = hypoth.anchors.kim_2019.1,
    refdata = combined_sct$k_tree,
    dims = 1:npcs)

hypoth.query.kim_2019.1 &lt;-
  MapQuery(
    anchorset = hypoth.anchors.kim_2019.1,
    reference = combined_sct,
    query = hypoth.query.kim_2019.1,
    refdata = list(k_tree = &quot;k_tree&quot;),
    reference.reduction = &quot;pca&quot;,
    reduction.model = &quot;umap&quot;)</code></pre>
<pre class="r"><code>p.kim_2019.1 &lt;-
  DimPlot(
    hypoth.query.kim_2019.1,
    reduction = &quot;ref.umap&quot;,
    group.by = &quot;predicted.k_tree&quot;,
    label = TRUE,
    label.size = 3,
    repel = TRUE) +
  NoLegend() +
  ggtitle(&quot;Query transferred labels&quot;)

p1 + p.kim_2019.1</code></pre>
<p><img src="figure/2022-12-12_pr.Rmd/pl-transfer-ageing-kim-2019-1-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="liu-et-al.-2022-10xv3" class="section level3">
<h3>Liu et al. 2022 10xV3</h3>
<pre class="r"><code>hypoth.query.liu_2022.1 &lt;-
  hypoth.list[[&quot;Anderson10x_batch_1&quot;]]
hypoth.anchors.liu_2022.1 &lt;-
  FindTransferAnchors(
    reference = combined_sct,
    query = hypoth.query.liu_2022.1,
    normalization.method = &quot;SCT&quot;,
    dims = 1:npcs,
    k.anchor = 10,
    k.score = 50,
    k.filter = 100,
    max.features = 500,
    n.trees = 100,
    reference.reduction = &quot;pca&quot;)

predictions.liu_2022.1 &lt;-
  TransferData(
    anchorset = hypoth.anchors.liu_2022.1,
    refdata = combined_sct$k_tree,
    dims = 1:npcs)

hypoth.query.liu_2022.1 &lt;-
  MapQuery(
    anchorset = hypoth.anchors.liu_2022.1,
    reference = combined_sct,
    query = hypoth.query.liu_2022.1,
    refdata = list(k_tree = &quot;k_tree&quot;),
    reference.reduction = &quot;pca&quot;,
    reduction.model = &quot;umap&quot;)</code></pre>
<pre class="r"><code>p.liu_2022.1 &lt;-
  DimPlot(
    hypoth.query.liu_2022.1,
    reduction = &quot;ref.umap&quot;,
    group.by = &quot;predicted.k_tree&quot;,
    label = TRUE,
    label.size = 3,
    repel = TRUE) +
  NoLegend() +
  ggtitle(&quot;Query transferred labels&quot;)

p1 + p.affinati_2021.1</code></pre>
<p><img src="figure/2022-12-12_pr.Rmd/pl-transfer-ageing-liu-2022-1-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="ventroposterior-hypothalamus" class="section level2">
<h2>Ventroposterior hypothalamus</h2>
<div id="mickelsen-et-al.-2020-10xv2" class="section level3">
<h3>Mickelsen et al. 2020 10xV2</h3>
<pre class="r"><code>hypoth.query.mickelsen_2020.1 &lt;-
  hypoth.list[[&quot;Flynn10x_batch_1&quot;]]
hypoth.anchors.mickelsen_2020.1 &lt;-
  FindTransferAnchors(
    reference = combined_sct,
    query = hypoth.query.mickelsen_2020.1,
    normalization.method = &quot;SCT&quot;,
    dims = 1:npcs,
    k.anchor = 10,
    k.score = 50,
    k.filter = 100,
    max.features = 500,
    n.trees = 100,
    reference.reduction = &quot;pca&quot;)

predictions.mickelsen_2020.1 &lt;-
  TransferData(
    anchorset = hypoth.anchors.mickelsen_2020.1,
    refdata = combined_sct$k_tree,
    dims = 1:npcs)

hypoth.query.mickelsen_2020.1 &lt;-
  MapQuery(
    anchorset = hypoth.anchors.mickelsen_2020.1,
    reference = combined_sct,
    query = hypoth.query.mickelsen_2020.1,
    refdata = list(k_tree = &quot;k_tree&quot;),
    reference.reduction = &quot;pca&quot;,
    reduction.model = &quot;umap&quot;)</code></pre>
<pre class="r"><code>p.mickelsen_2020.1 &lt;-
  DimPlot(
    hypoth.query.mickelsen_2020.1,
    reduction = &quot;ref.umap&quot;,
    group.by = &quot;predicted.k_tree&quot;,
    label = TRUE,
    label.size = 3,
    repel = TRUE) +
  NoLegend() +
  ggtitle(&quot;Query transferred labels&quot;)

p1 + p.mickelsen_2020.1</code></pre>
<p><img src="figure/2022-12-12_pr.Rmd/pl-transfer-ageing-mickelsen-2020-1-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="preoptic-area" class="section level2">
<h2>Preoptic area</h2>
<div id="moffitt-et-al.-2018-10xv2" class="section level3">
<h3>Moffitt et al. 2018 10xV2</h3>
<pre class="r"><code>hypoth.query.moffitt_2018.1 &lt;-
  hypoth.list[[&quot;Moffit10x_batch_1&quot;]]
hypoth.anchors.moffitt_2018.1 &lt;-
  FindTransferAnchors(
    reference = combined_sct,
    query = hypoth.query.moffitt_2018.1,
    normalization.method = &quot;SCT&quot;,
    dims = 1:npcs,
    k.anchor = 10,
    k.score = 50,
    k.filter = 100,
    max.features = 500,
    n.trees = 100,
    reference.reduction = &quot;pca&quot;)

predictions.moffitt_2018.1 &lt;-
  TransferData(
    anchorset = hypoth.anchors.moffitt_2018.1,
    refdata = combined_sct$k_tree,
    dims = 1:npcs)

hypoth.query.moffitt_2018.1 &lt;-
  MapQuery(
    anchorset = hypoth.anchors.moffitt_2018.1,
    reference = combined_sct,
    query = hypoth.query.moffitt_2018.1,
    refdata = list(k_tree = &quot;k_tree&quot;),
    reference.reduction = &quot;pca&quot;,
    reduction.model = &quot;umap&quot;)</code></pre>
<pre class="r"><code>p.moffitt_2018.1 &lt;-
  DimPlot(
    hypoth.query.moffitt_2018.1,
    reduction = &quot;ref.umap&quot;,
    group.by = &quot;predicted.k_tree&quot;,
    label = TRUE,
    label.size = 3,
    repel = TRUE) +
  NoLegend() +
  ggtitle(&quot;Query transferred labels&quot;)

p1 + p.moffitt_2018.1</code></pre>
<p><img src="figure/2022-12-12_pr.Rmd/pl-transfer-ageing-moffitt-2018-1-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>hypoth.query.moffitt_2018.2 &lt;-
  hypoth.list[[&quot;Moffit10x_batch_2&quot;]]
hypoth.anchors.moffitt_2018.2 &lt;-
  FindTransferAnchors(
    reference = combined_sct,
    query = hypoth.query.moffitt_2018.2,
    normalization.method = &quot;SCT&quot;,
    dims = 1:npcs,
    k.anchor = 10,
    k.score = 50,
    k.filter = 100,
    max.features = 500,
    n.trees = 100,
    reference.reduction = &quot;pca&quot;)

predictions.moffitt_2018.2 &lt;-
  TransferData(
    anchorset = hypoth.anchors.moffitt_2018.2,
    refdata = combined_sct$k_tree,
    dims = 1:npcs)

hypoth.query.moffitt_2018.2 &lt;-
  MapQuery(
    anchorset = hypoth.anchors.moffitt_2018.2,
    reference = combined_sct,
    query = hypoth.query.moffitt_2018.2,
    refdata = list(k_tree = &quot;k_tree&quot;),
    reference.reduction = &quot;pca&quot;,
    reduction.model = &quot;umap&quot;)</code></pre>
<pre class="r"><code>p.moffitt_2018.2 &lt;-
  DimPlot(
    hypoth.query.moffitt_2018.2,
    reduction = &quot;ref.umap&quot;,
    group.by = &quot;predicted.k_tree&quot;,
    label = TRUE,
    label.size = 3,
    repel = TRUE) +
  NoLegend() +
  ggtitle(&quot;Query transferred labels&quot;)

p1 + p.moffitt_2018.2</code></pre>
<p><img src="figure/2022-12-12_pr.Rmd/pl-transfer-ageing-moffitt-2018-2-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>hypoth.query.moffitt_2018.3 &lt;-
  hypoth.list[[&quot;Moffit10x_batch_3&quot;]]
hypoth.anchors.moffitt_2018.3 &lt;-
  FindTransferAnchors(
    reference = combined_sct,
    query = hypoth.query.moffitt_2018.3,
    normalization.method = &quot;SCT&quot;,
    dims = 1:npcs,
    k.anchor = 10,
    k.score = 50,
    k.filter = 100,
    max.features = 500,
    n.trees = 100,
    reference.reduction = &quot;pca&quot;)

predictions.moffitt_2018.3 &lt;-
  TransferData(
    anchorset = hypoth.anchors.moffitt_2018.3,
    refdata = combined_sct$k_tree,
    dims = 1:npcs)

hypoth.query.moffitt_2018.3 &lt;-
  MapQuery(
    anchorset = hypoth.anchors.moffitt_2018.3,
    reference = combined_sct,
    query = hypoth.query.moffitt_2018.3,
    refdata = list(k_tree = &quot;k_tree&quot;),
    reference.reduction = &quot;pca&quot;,
    reduction.model = &quot;umap&quot;)</code></pre>
<pre class="r"><code>p.moffitt_2018.3 &lt;-
  DimPlot(
    hypoth.query.moffitt_2018.3,
    reduction = &quot;ref.umap&quot;,
    group.by = &quot;predicted.k_tree&quot;,
    label = TRUE,
    label.size = 3,
    repel = TRUE) +
  NoLegend() +
  ggtitle(&quot;Query transferred labels&quot;)

p1 + p.moffitt_2018.3</code></pre>
<p><img src="figure/2022-12-12_pr.Rmd/pl-transfer-ageing-moffitt-2018-3-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="suprachiasmatic-nucleus" class="section level2">
<h2>Suprachiasmatic nucleus</h2>
<div id="wen-et-al.-2020-10xv2" class="section level3">
<h3>Wen et al. 2020 10xV2</h3>
<pre class="r"><code>hypoth.query.wen_2020.1 &lt;-
  hypoth.list[[&quot;wen10x_batch_1&quot;]]
hypoth.anchors.wen_2020.1 &lt;-
  FindTransferAnchors(
    reference = combined_sct,
    query = hypoth.query.wen_2020.1,
    normalization.method = &quot;SCT&quot;,
    dims = 1:npcs,
    k.anchor = 10,
    k.score = 50,
    k.filter = 100,
    max.features = 500,
    n.trees = 100,
    reference.reduction = &quot;pca&quot;)

predictions.wen_2020.1 &lt;-
  TransferData(
    anchorset = hypoth.anchors.wen_2020.1,
    refdata = combined_sct$k_tree,
    dims = 1:npcs)

hypoth.query.wen_2020.1 &lt;-
  MapQuery(
    anchorset = hypoth.anchors.wen_2020.1,
    reference = combined_sct,
    query = hypoth.query.wen_2020.1,
    refdata = list(k_tree = &quot;k_tree&quot;),
    reference.reduction = &quot;pca&quot;,
    reduction.model = &quot;umap&quot;)</code></pre>
<pre class="r"><code>p.wen_2020.1 &lt;-
  DimPlot(
    hypoth.query.wen_2020.1,
    reduction = &quot;ref.umap&quot;,
    group.by = &quot;predicted.k_tree&quot;,
    label = TRUE,
    label.size = 3,
    repel = TRUE) +
  NoLegend() +
  ggtitle(&quot;Query transferred labels&quot;)

p1 + p.wen_2020.1</code></pre>
<p><img src="figure/2022-12-12_pr.Rmd/pl-transfer-ageing-wen-2020-1-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="wen-et-al.-2020-dropseq" class="section level3">
<h3>Wen et al. 2020 Dropseq</h3>
<pre class="r"><code>hypoth.query.wen_2020.2 &lt;-
  hypoth.list[[&quot;wenDropseq_batch_1&quot;]]
hypoth.anchors.wen_2020.2 &lt;-
  FindTransferAnchors(
    reference = combined_sct,
    query = hypoth.query.wen_2020.2,
    normalization.method = &quot;SCT&quot;,
    dims = 1:npcs,
    k.anchor = 10,
    k.score = 50,
    k.filter = 100,
    max.features = 500,
    n.trees = 100,
    reference.reduction = &quot;pca&quot;)

predictions.wen_2020.2 &lt;-
  TransferData(
    anchorset = hypoth.anchors.wen_2020.2,
    refdata = combined_sct$k_tree,
    dims = 1:npcs)

hypoth.query.wen_2020.2 &lt;-
  MapQuery(
    anchorset = hypoth.anchors.wen_2020.2,
    reference = combined_sct,
    query = hypoth.query.wen_2020.2,
    refdata = list(k_tree = &quot;k_tree&quot;),
    reference.reduction = &quot;pca&quot;,
    reduction.model = &quot;umap&quot;)</code></pre>
<pre class="r"><code>p.wen_2020.2 &lt;-
  DimPlot(
    hypoth.query.wen_2020.2,
    reduction = &quot;ref.umap&quot;,
    group.by = &quot;predicted.k_tree&quot;,
    label = TRUE,
    label.size = 3,
    repel = TRUE) +
  NoLegend() +
  ggtitle(&quot;Query transferred labels&quot;)

p1 + p.wen_2020.2</code></pre>
<p><img src="figure/2022-12-12_pr.Rmd/pl-transfer-ageing-wen-2020-2-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="morris-et-al.-2021-10xv3" class="section level3">
<h3>Morris et al. 2021 10xV3</h3>
<pre class="r"><code>hypoth.query.morris_2021.1 &lt;-
  hypoth.list[[&quot;Morris10x_batch_1&quot;]]
hypoth.anchors.morris_2021.1 &lt;-
  FindTransferAnchors(
    reference = combined_sct,
    query = hypoth.query.morris_2021.1,
    normalization.method = &quot;SCT&quot;,
    dims = 1:npcs,
    k.anchor = 10,
    k.score = 50,
    k.filter = 100,
    max.features = 500,
    n.trees = 100,
    reference.reduction = &quot;pca&quot;)

predictions.morris_2021.1 &lt;-
  TransferData(
    anchorset = hypoth.anchors.morris_2021.1,
    refdata = combined_sct$k_tree,
    dims = 1:npcs)

hypoth.query.morris_2021.1 &lt;-
  MapQuery(
    anchorset = hypoth.anchors.morris_2021.1,
    reference = combined_sct,
    query = hypoth.query.morris_2021.1,
    refdata = list(k_tree = &quot;k_tree&quot;),
    reference.reduction = &quot;pca&quot;,
    reduction.model = &quot;umap&quot;)</code></pre>
<pre class="r"><code>p.morris_2021.1 &lt;-
  DimPlot(
    hypoth.query.morris_2021.1,
    reduction = &quot;ref.umap&quot;,
    group.by = &quot;predicted.k_tree&quot;,
    label = TRUE,
    label.size = 3,
    repel = TRUE) +
  NoLegend() +
  ggtitle(&quot;Query transferred labels&quot;)

p1 + p.morris_2021.1</code></pre>
<p><img src="figure/2022-12-12_pr.Rmd/pl-transfer-ageing-morris-2021-1-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>hypoth.query.morris_2021.2 &lt;-
  hypoth.list[[&quot;Morris10x_batch_2&quot;]]
hypoth.anchors.morris_2021.2 &lt;-
  FindTransferAnchors(
    reference = combined_sct,
    query = hypoth.query.morris_2021.2,
    normalization.method = &quot;SCT&quot;,
    dims = 1:npcs,
    k.anchor = 10,
    k.score = 50,
    k.filter = 100,
    max.features = 500,
    n.trees = 100,
    reference.reduction = &quot;pca&quot;)

predictions.morris_2021.2 &lt;-
  TransferData(
    anchorset = hypoth.anchors.morris_2021.2,
    refdata = combined_sct$k_tree,
    dims = 1:npcs)

hypoth.query.morris_2021.2 &lt;-
  MapQuery(
    anchorset = hypoth.anchors.morris_2021.2,
    reference = combined_sct,
    query = hypoth.query.morris_2021.2,
    refdata = list(k_tree = &quot;k_tree&quot;),
    reference.reduction = &quot;pca&quot;,
    reduction.model = &quot;umap&quot;)</code></pre>
<pre class="r"><code>p.morris_2021.2 &lt;-
  DimPlot(
    hypoth.query.morris_2021.2,
    reduction = &quot;ref.umap&quot;,
    group.by = &quot;predicted.k_tree&quot;,
    label = TRUE,
    label.size = 3,
    repel = TRUE) +
  NoLegend() +
  ggtitle(&quot;Query transferred labels&quot;)

p1 + p.morris_2021.2</code></pre>
<p><img src="figure/2022-12-12_pr.Rmd/pl-transfer-ageing-morris-2021-2-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>hypoth.query.morris_2021.3 &lt;-
  hypoth.list[[&quot;Morris10x_batch_3&quot;]]
hypoth.anchors.morris_2021.3 &lt;-
  FindTransferAnchors(
    reference = combined_sct,
    query = hypoth.query.morris_2021.3,
    normalization.method = &quot;SCT&quot;,
    dims = 1:npcs,
    k.anchor = 10,
    k.score = 50,
    k.filter = 100,
    max.features = 500,
    n.trees = 100,
    reference.reduction = &quot;pca&quot;)

predictions.morris_2021.3 &lt;-
  TransferData(
    anchorset = hypoth.anchors.morris_2021.3,
    refdata = combined_sct$k_tree,
    dims = 1:npcs)

hypoth.query.morris_2021.3 &lt;-
  MapQuery(
    anchorset = hypoth.anchors.morris_2021.3,
    reference = combined_sct,
    query = hypoth.query.morris_2021.3,
    refdata = list(k_tree = &quot;k_tree&quot;),
    reference.reduction = &quot;pca&quot;,
    reduction.model = &quot;umap&quot;)</code></pre>
<pre class="r"><code>p.morris_2021.3 &lt;-
  DimPlot(
    hypoth.query.morris_2021.3,
    reduction = &quot;ref.umap&quot;,
    group.by = &quot;predicted.k_tree&quot;,
    label = TRUE,
    label.size = 3,
    repel = TRUE) +
  NoLegend() +
  ggtitle(&quot;Query transferred labels&quot;)

p1 + p.morris_2021.3</code></pre>
<p><img src="figure/2022-12-12_pr.Rmd/pl-transfer-ageing-morris-2021-3-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="lateral-hypothalamic-area" class="section level2">
<h2>Lateral hypothalamic area</h2>
<div id="mickelsen-et-al.-2019-10xv2" class="section level3">
<h3>Mickelsen et al. 2019 10xV2</h3>
<pre class="r"><code>hypoth.query.mickelsen_2019.1 &lt;-
  hypoth.list[[&quot;Mickelsen10x_batch_1&quot;]]
hypoth.anchors.mickelsen_2019.1 &lt;-
  FindTransferAnchors(
    reference = combined_sct,
    query = hypoth.query.mickelsen_2019.1,
    normalization.method = &quot;SCT&quot;,
    dims = 1:npcs,
    k.anchor = 10,
    k.score = 50,
    k.filter = 100,
    max.features = 500,
    n.trees = 100,
    reference.reduction = &quot;pca&quot;)

predictions.mickelsen_2019.1 &lt;-
  TransferData(
    anchorset = hypoth.anchors.mickelsen_2019.1,
    refdata = combined_sct$k_tree,
    dims = 1:npcs)

hypoth.query.mickelsen_2019.1 &lt;-
  MapQuery(
    anchorset = hypoth.anchors.mickelsen_2019.1,
    reference = combined_sct,
    query = hypoth.query.mickelsen_2019.1,
    refdata = list(k_tree = &quot;k_tree&quot;),
    reference.reduction = &quot;pca&quot;,
    reduction.model = &quot;umap&quot;)</code></pre>
<pre class="r"><code>p.mickelsen_2019.1 &lt;-
  DimPlot(
    hypoth.query.mickelsen_2019.1,
    reduction = &quot;ref.umap&quot;,
    group.by = &quot;predicted.k_tree&quot;,
    label = TRUE,
    label.size = 3,
    repel = TRUE) +
  NoLegend() +
  ggtitle(&quot;Query transferred labels&quot;)

p1 + p.mickelsen_2019.1</code></pre>
<p><img src="figure/2022-12-12_pr.Rmd/pl-transfer-ageing-mickelsen-2019-1-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="rossi-et-al.-2019-dropseq" class="section level3">
<h3>Rossi et al. 2019 Dropseq</h3>
<pre class="r"><code>hypoth.query.rossi_2019.1 &lt;-
  hypoth.list[[&quot;Mickelsen10x_batch_1&quot;]]
hypoth.anchors.rossi_2019.1 &lt;-
  FindTransferAnchors(
    reference = combined_sct,
    query = hypoth.query.rossi_2019.1,
    normalization.method = &quot;SCT&quot;,
    dims = 1:npcs,
    k.anchor = 10,
    k.score = 50,
    k.filter = 100,
    max.features = 500,
    n.trees = 100,
    reference.reduction = &quot;pca&quot;)

predictions.rossi_2019.1 &lt;-
  TransferData(
    anchorset = hypoth.anchors.rossi_2019.1,
    refdata = combined_sct$k_tree,
    dims = 1:npcs)

hypoth.query.rossi_2019.1 &lt;-
  MapQuery(
    anchorset = hypoth.anchors.rossi_2019.1,
    reference = combined_sct,
    query = hypoth.query.rossi_2019.1,
    refdata = list(k_tree = &quot;k_tree&quot;),
    reference.reduction = &quot;pca&quot;,
    reduction.model = &quot;umap&quot;)</code></pre>
<pre class="r"><code>p.rossi_2019.1 &lt;-
  DimPlot(
    hypoth.query.rossi_2019.1,
    reduction = &quot;ref.umap&quot;,
    group.by = &quot;predicted.k_tree&quot;,
    label = TRUE,
    label.size = 3,
    repel = TRUE) +
  NoLegend() +
  ggtitle(&quot;Query transferred labels&quot;)

p1 + p.rossi_2019.1</code></pre>
<p><img src="figure/2022-12-12_pr.Rmd/pl-transfer-ageing-rossi-2019-1-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="paraventricular-nucleus" class="section level3">
<h3>Paraventricular nucleus</h3>
<pre class="r"><code># hypoth.query.rupp_2021.1 &lt;-
#   hypoth.list[[&quot;Rupp10x_batch_1&quot;]]
# hypoth.anchors.rupp_2021.1 &lt;-
#   FindTransferAnchors(
#     reference = combined_sct,
#     query = hypoth.query.rupp_2021.1,
#     normalization.method = &quot;SCT&quot;,
#     dims = 1:npcs,
#     k.anchor = 10,
#     k.score = 50,
#     k.filter = 100,
#     max.features = 500,
#     n.trees = 100,
#     reference.reduction = &quot;pca&quot;)
# 
# predictions.rupp_2021.1 &lt;-
#   TransferData(
#     anchorset = hypoth.anchors.rupp_2021.1,
#     refdata = combined_sct$k_tree,
#     dims = 1:npcs)
# 
# hypoth.query.rupp_2021.1 &lt;-
#   MapQuery(
#     anchorset = hypoth.anchors.rupp_2021.1,
#     reference = combined_sct,
#     query = hypoth.query.rupp_2021.1,
#     refdata = list(k_tree = &quot;k_tree&quot;),
#     reference.reduction = &quot;pca&quot;,
#     reduction.model = &quot;umap&quot;)</code></pre>
<pre class="r"><code># p.rupp_2021.1 &lt;-
#   DimPlot(
#     hypoth.query.rupp_2021.1,
#     reduction = &quot;ref.umap&quot;,
#     group.by = &quot;predicted.k_tree&quot;,
#     label = TRUE,
#     label.size = 3,
#     repel = TRUE) +
#   NoLegend() +
#   ggtitle(&quot;Query transferred labels&quot;)
# 
# p1 + p.rupp_2021.1</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.2 (2022-10-31)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 22.04.1 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3
LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods  
[7] base     

other attached packages:
 [1] ggraph_2.1.0.9000      mrtree_0.0.0.9000     
 [3] RobustRankAggreg_1.2.1 biomaRt_2.54.0        
 [5] scCustomize_1.0.2      Scillus_0.5.0         
 [7] Nebulosa_1.8.0         ggplot2_3.4.0         
 [9] swne_0.6.20            patchwork_1.1.2.9000  
[11] UpSetR_1.4.0           SeuratDisk_0.0.0.9020 
[13] SeuratWrappers_0.3.1   SeuratObject_4.1.3    
[15] Seurat_4.3.0           reticulate_1.26       
[17] kableExtra_1.3.4       future_1.29.0         
[19] skimr_2.1.4            stringr_1.5.0.9000    
[21] magrittr_2.0.3         dplyr_1.0.10          
[23] purrr_0.3.5            readr_2.1.3           
[25] here_1.0.1             workflowr_1.7.0       

loaded via a namespace (and not attached):
  [1] rsvd_1.0.5                  ica_1.0-3                  
  [3] svglite_2.1.0               ps_1.7.2                   
  [5] foreach_1.5.2               lmtest_0.9-40              
  [7] rprojroot_2.0.3             crayon_1.5.2               
  [9] MASS_7.3-58.1               nlme_3.1-160               
 [11] backports_1.4.1             rlang_1.0.6                
 [13] XVector_0.38.0              ROCR_1.0-11                
 [15] irlba_2.3.5.1               callr_3.7.3                
 [17] limma_3.54.0                filelock_1.0.2             
 [19] data.tree_1.0.0             rjson_0.2.21               
 [21] bit64_4.0.5                 glue_1.6.2                 
 [23] sctransform_0.3.5           parallel_4.2.2             
 [25] processx_3.8.0              vipor_0.4.5                
 [27] spatstat.sparse_3.0-0       AnnotationDbi_1.60.0       
 [29] BiocGenerics_0.44.0         spatstat.geom_3.0-3        
 [31] tidyselect_1.2.0            SummarizedExperiment_1.28.0
 [33] liger_2.0.1                 fitdistrplus_1.1-8         
 [35] XML_3.99-0.13               tidyr_1.2.1.9001           
 [37] zoo_1.8-11                  ggpubr_0.5.0.999           
 [39] xtable_1.8-4                formattable_0.2.1          
 [41] evaluate_0.18               cli_3.4.1                  
 [43] zlibbioc_1.44.0             rstudioapi_0.14            
 [45] miniUI_0.1.1.1              sp_1.5-1                   
 [47] whisker_0.4.1               bslib_0.4.1                
 [49] fastmatch_1.1-3             treeio_1.23.0              
 [51] maps_3.4.1                  shiny_1.7.3                
 [53] xfun_0.35                   askpass_1.1                
 [55] clue_0.3-63                 usedist_0.4.0              
 [57] cluster_2.1.4               tidygraph_1.2.2            
 [59] clusterGeneration_1.3.7     KEGGREST_1.38.0            
 [61] expm_0.999-6                SymSim_0.0.0.9000          
 [63] tibble_3.1.8                ggrepel_0.9.2.9999         
 [65] ape_5.6-2                   dendextend_1.16.0          
 [67] listenv_0.8.0               Biostrings_2.66.0          
 [69] png_0.1-8                   withr_2.5.0                
 [71] ggforce_0.4.1               bitops_1.0-7               
 [73] plyr_1.8.8                  coda_0.19-4                
 [75] pracma_2.4.2                pillar_1.8.1               
 [77] GlobalOptions_0.1.2         cachem_1.0.6               
 [79] fs_1.5.2                    scatterplot3d_0.3-42       
 [81] hdf5r_1.3.7                 GetoptLong_1.0.5           
 [83] paletteer_1.5.0             vctrs_0.5.1                
 [85] ellipsis_0.3.2              generics_0.1.3             
 [87] tools_4.2.2                 beeswarm_0.4.0             
 [89] tweenr_2.0.2                munsell_0.5.0              
 [91] proxy_0.4-27                DelayedArray_0.24.0        
 [93] fastmap_1.1.0               compiler_4.2.2             
 [95] abind_1.4-5                 httpuv_1.6.6               
 [97] ggimage_0.3.1               plotly_4.10.1              
 [99] GenomeInfoDbData_1.2.9      gridExtra_2.3              
[101] colorway_0.2.0              lattice_0.20-45            
[103] deldir_1.0-6                snow_0.4-4                 
[105] utf8_1.2.2                  later_1.3.0                
[107] BiocFileCache_2.6.0         jsonlite_1.8.4             
[109] scales_1.2.1                tidytree_0.4.1             
[111] pbapply_1.6-0               carData_3.0-5              
[113] lazyeval_0.2.2              promises_1.2.0.1           
[115] car_3.1-1                   doParallel_1.0.17          
[117] R.utils_2.12.2              goftest_1.2-3              
[119] checkmate_2.1.0             spatstat.utils_3.0-1       
[121] rmarkdown_2.18              cowplot_1.1.1              
[123] textshaping_0.3.6           webshot_0.5.4              
[125] Rtsne_0.16                  glmGamPoi_1.10.0           
[127] forcats_0.5.2               Biobase_2.58.0             
[129] uwot_0.1.14                 igraph_1.3.5               
[131] plotrix_3.8-2               numDeriv_2016.8-1.1        
[133] survival_3.4-0              yaml_2.3.6                 
[135] systemfonts_1.0.4           htmltools_0.5.4            
[137] memoise_2.0.1               graphlayouts_0.8.4         
[139] quadprog_1.5-8              IRanges_2.32.0             
[141] viridisLite_0.4.1           digest_0.6.31              
[143] assertthat_0.2.1            mime_0.12                  
[145] rappdirs_0.3.3              repr_1.1.4                 
[147] RSQLite_2.2.19              yulab.utils_0.0.5          
[149] future.apply_1.10.0         ggmin_0.0.0.9000           
[151] remotes_2.4.2               data.table_1.14.6          
[153] blob_1.2.3                  S4Vectors_0.36.1           
[155] R.oo_1.25.0                 ragg_1.2.4                 
[157] splines_4.2.2               labeling_0.4.2             
[159] rematch2_2.1.2              RCurl_1.98-1.9             
[161] broom_1.0.1                 ks_1.14.0                  
[163] hms_1.1.2                   colorspace_2.0-3           
[165] base64enc_0.1-3             mnormt_2.1.1               
[167] BiocManager_1.30.19         ggbeeswarm_0.6.0           
[169] GenomicRanges_1.30.3        shape_1.4.6                
[171] aplot_0.1.9                 sass_0.4.4                 
[173] Rcpp_1.0.9                  mclust_6.0.0               
[175] RANN_2.6.1                  mvtnorm_1.1-3              
[177] circlize_0.4.15             NNLM_0.4.4                 
[179] fansi_1.0.3                 tzdb_0.3.0                 
[181] parallelly_1.32.1           R6_2.5.1                   
[183] grid_4.2.2                  phytools_1.2-0             
[185] ggridges_0.5.4              lifecycle_1.0.3            
[187] curl_4.3.3                  ggsignif_0.6.4             
[189] phangorn_2.10.0             leiden_0.4.3               
[191] jquerylib_0.1.4             snakecase_0.11.0           
[193] Matrix_1.5-3                RcppAnnoy_0.0.20           
[195] RColorBrewer_1.1-3          iterators_1.0.14           
[197] spatstat.explore_3.0-5      htmlwidgets_1.5.4          
[199] umap_0.2.9.0                polyclip_1.10-4            
[201] gridGraphics_0.5-1          timechange_0.1.1           
[203] optimParallel_1.0-2         rvest_1.0.3                
[205] ComplexHeatmap_2.14.0       mgcv_1.8-41                
[207] globals_0.16.2              openssl_2.0.5              
[209] spatstat.random_3.0-1       progressr_0.12.0           
[211] codetools_0.2-18            matrixStats_0.63.0         
[213] lubridate_1.9.0             FNN_1.1.3.1                
[215] getPass_0.2-2               prettyunits_1.1.1          
[217] SingleCellExperiment_1.20.0 dbplyr_2.2.1               
[219] RSpectra_0.16-1             R.methodsS3_1.8.2          
[221] GenomeInfoDb_1.34.4         gtable_0.3.1               
[223] DBI_1.1.3                   git2r_0.30.1               
[225] stats4_4.2.2                ggfun_0.0.9                
[227] tensor_1.5                  httr_1.4.4                 
[229] highr_0.9                   KernSmooth_2.23-20         
[231] stringi_1.7.8               vroom_1.6.0                
[233] progress_1.2.2              reshape2_1.4.4             
[235] farver_2.1.1                viridis_0.6.2              
[237] ggtree_3.7.1.002            magick_2.7.3               
[239] combinat_0.0-8              xml2_1.3.3                 
[241] ggplotify_0.1.0             scattermore_0.8            
[243] bit_4.0.5                   clustree_0.5.0             
[245] MatrixGenerics_1.10.0       spatstat.data_3.0-0        
[247] janitor_2.1.0.9000          pkgconfig_2.0.3            
[249] ggprism_1.0.4               rstatix_0.7.1.999          
[251] knitr_1.41                 </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
