<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Evgenii Tretiakov" />


<title>Regional and functional heterogeneity of hypothalamic astrocytes</title>

<script src="site_libs/header-attrs-2.20/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">heteroAstrocytes</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="hfd.html">HFD</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Regional and functional heterogeneity of
hypothalamic astrocytes</h1>
<h4 class="author">Evgenii Tretiakov</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-03-23
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>1_heteroAstrocytes/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R
Markdown file created these results, you’ll want to first commit it to
the Git repo. If you’re still working on the analysis, you can ignore
this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220624code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20220624)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220624code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220624)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomEugOTheteroAstrocytestree9e657bfe05bc16846c92b75373c11f168a391d75targetblank9e657bfa">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/EugOT/heteroAstrocytes/tree/9e657bfe05bc16846c92b75373c11f168a391d75" target="_blank">9e657bf</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomEugOTheteroAstrocytestree9e657bfe05bc16846c92b75373c11f168a391d75targetblank9e657bfa"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/EugOT/heteroAstrocytes/tree/9e657bfe05bc16846c92b75373c11f168a391d75" target="_blank">9e657bf</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.Rhistory
    Ignored:    analysis/transfers.RData
    Ignored:    data/astrocytes-srt_by_nuclei.bak.rds
    Ignored:    data/comb_astrocytes.bak.h5Seurat
    Ignored:    data/comb_astrocytes.bak.h5ad
    Ignored:    data/deng2019_arc_chow.h5Seurat
    Ignored:    data/deng2019_arc_chow.h5ad
    Ignored:    data/deng2019_arc_chow_astrocytes.h5Seurat
    Ignored:    data/deng2019_arc_chow_astrocytes.h5ad
    Ignored:    data/deng2019_arc_chow_astrocytes_fin.h5Seurat
    Ignored:    data/deng2019_arc_chow_astrocytes_fin.h5ad
    Ignored:    data/deng2019_arc_chow_clusters.h5Seurat
    Ignored:    data/deng2019_arc_chow_clusters.h5ad
    Ignored:    data/deng2019_arc_hfd.h5Seurat
    Ignored:    data/deng2019_arc_hfd.h5ad
    Ignored:    data/deng2019_arc_hfd_astrocytes.h5Seurat
    Ignored:    data/deng2019_arc_hfd_astrocytes.h5ad
    Ignored:    data/deng2019_arc_hfd_astrocytes_fin.h5Seurat
    Ignored:    data/deng2019_arc_hfd_astrocytes_fin.h5ad
    Ignored:    data/deng2019_arc_hfd_clusters.h5Seurat
    Ignored:    data/deng2019_arc_hfd_clusters.h5ad
    Ignored:    data/deng2019_arc_refi_astrocytes.h5Seurat
    Ignored:    data/deng2019_arc_refi_astrocytes.h5ad
    Ignored:    data/deng_2020_arc_chow.h5Seurat
    Ignored:    data/deng_2020_arc_chow.h5ad
    Ignored:    data/deng_2020_arc_chow_astro-npy-pomc.h5Seurat
    Ignored:    data/deng_2020_arc_chow_astro-npy-pomc.h5ad
    Ignored:    data/deng_2020_arc_chow_astrocytes.h5Seurat
    Ignored:    data/deng_2020_arc_chow_astrocytes.h5ad
    Ignored:    data/deng_2020_arc_chow_astrocytes_fin.h5Seurat
    Ignored:    data/deng_2020_arc_chow_astrocytes_fin.h5ad
    Ignored:    data/deng_2020_arc_chow_astrocytes_sct.h5Seurat
    Ignored:    data/deng_2020_arc_chow_astrocytes_sct.h5ad
    Ignored:    data/deng_2020_arc_chow_clusters.h5Seurat
    Ignored:    data/deng_2020_arc_chow_clusters.h5ad
    Ignored:    data/deng_2020_arc_hfd.h5Seurat
    Ignored:    data/deng_2020_arc_hfd.h5ad
    Ignored:    data/deng_2020_arc_hfd_astrocytes.h5Seurat
    Ignored:    data/deng_2020_arc_hfd_astrocytes.h5ad
    Ignored:    data/deng_2020_arc_hfd_astrocytes_fin.h5Seurat
    Ignored:    data/deng_2020_arc_hfd_astrocytes_fin.h5ad
    Ignored:    data/deng_2020_arc_hfd_clusters.h5Seurat
    Ignored:    data/deng_2020_arc_hfd_clusters.h5ad
    Ignored:    data/deng_2020_arc_refi_astrocytes.h5Seurat
    Ignored:    data/deng_2020_arc_refi_astrocytes.h5ad
    Ignored:    data/hypoth.query.affinati_2021.1.h5Seurat
    Ignored:    data/hypoth.query.affinati_2021.3.h5Seurat
    Ignored:    data/hypoth.query.affinati_2021.4.h5Seurat
    Ignored:    data/hypoth.query.campbell_2017.1.h5Seurat
    Ignored:    data/hypoth.query.deng_2021.1.h5Seurat
    Ignored:    data/hypoth.query.deng_2021.3.h5Seurat
    Ignored:    data/hypoth.query.deng_2021.5.h5Seurat
    Ignored:    data/hypoth.query.kim_2019.1.h5Seurat
    Ignored:    data/hypoth.query.liu_2022.1.h5Seurat
    Ignored:    data/hypoth.query.lopez_2021.1.h5Seurat
    Ignored:    data/hypoth.query.lutomska_2022.1.h5Seurat
    Ignored:    data/hypoth.query.mickelsen_2019.1.h5Seurat
    Ignored:    data/hypoth.query.mickelsen_2020.1.h5Seurat
    Ignored:    data/hypoth.query.moffitt_2018.1.h5Seurat
    Ignored:    data/hypoth.query.moffitt_2018.2.h5Seurat
    Ignored:    data/hypoth.query.moffitt_2018.3.h5Seurat
    Ignored:    data/hypoth.query.morris_2021.1.h5Seurat
    Ignored:    data/hypoth.query.morris_2021.2.h5Seurat
    Ignored:    data/hypoth.query.morris_2021.3.h5Seurat
    Ignored:    data/hypoth.query.rossi_2019.1.h5Seurat
    Ignored:    data/hypoth.query.rupp_2021.1.h5Seurat
    Ignored:    data/hypoth.query.wen_2020.1.h5Seurat
    Ignored:    data/hypoth.query.wen_2020.2.h5Seurat
    Ignored:    data/integration_features_by_nuclei.bak.rds
    Ignored:    data/webb_astrocytes_combined_sct.h5Seurat
    Ignored:    transfers.RData

Untracked files:
    Untracked:  .idea/
    Untracked:  .ipynb_checkpoints/
    Untracked:  analysis/conv-integration-all-together-cloud.Rmd
    Untracked:  analysis/plot-conv-integration-all-together-cloud.Rmd
    Untracked:  data .ipynb
    Untracked:  data/.ipynb_checkpoints/
    Untracked:  data/astrocytes_X1_PRJNA453138_densmap_supervised.pdf
    Untracked:  data/astrocytes_X1_PRJNA515063_densmap_supervised.pdf
    Untracked:  data/astrocytes_X1_PRJNA548532_densmap_supervised.pdf
    Untracked:  data/astrocytes_X1_PRJNA604055_densmap_supervised.pdf
    Untracked:  data/astrocytes_X1_PRJNA611624_densmap_supervised.pdf
    Untracked:  data/astrocytes_X1_PRJNA679294_densmap_supervised.pdf
    Untracked:  data/astrocytes_X1_PRJNA705596_densmap_supervised.pdf
    Untracked:  data/astrocytes_X1_PRJNA722418_densmap_supervised.pdf
    Untracked:  data/astrocytes_X1_PRJNA723345_densmap_supervised.pdf
    Untracked:  data/astrocytes_X1_PRJNA798401_densmap_supervised.pdf
    Untracked:  data/astrocytes_X1_PRJNA815819_densmap_supervised.pdf
    Untracked:  data/astrocytes_X1_PRJNA847050_densmap_supervised.pdf
    Untracked:  data/data
    Untracked:  data/output/
    Untracked:  data/resolved-full_tree.ipynb
    Untracked:  data/resolved-min_tree-balanced_train.ipynb
    Untracked:  data/resolved-min_tree-microclusters.ipynb
    Untracked:  data/resolved-min_tree-subclusters.ipynb
    Untracked:  data/resolved-min_tree.ipynb
    Untracked:  data/resolved.ipynb
    Untracked:  output/tables/deng2021-chow-all-mrk_logreg-sct.csv
    Untracked:  output/tables/deng2021-chow-all-mrk_roc-sct-iter2.csv
    Untracked:  output/tables/deng2021-chow-all-mrk_roc-sct-iter3.csv
    Untracked:  output/tables/deng2021-chow-all-mrk_roc-sct.csv
    Untracked:  output/tables/deng2021-chow-all-mrk_wilcox-sct-iter2.csv
    Untracked:  output/tables/deng2021-chow-all-mrk_wilcox-sct-iter3.csv
    Untracked:  output/tables/deng2021-chow-all-mrk_wilcox-sct.csv
    Untracked:  output/tables/deng2021-hfd-all-mrk_roc-sct.csv
    Untracked:  output/tables/deng2021-hfd-all-mrk_wilcox-sct.csv
    Untracked:  poetry.lock
    Untracked:  pyproject.toml

Unstaged changes:
    Deleted:    analysis/2022-11-16_pr.Rmd
    Modified:   analysis/2022-12-12_pr.Rmd
    Modified:   analysis/hfd.Rmd
    Modified:   analysis/index.Rmd
    Deleted:    astro-heatmap.pdf
    Modified:   code/astrogenesrankaggr-temp.R
    Modified:   code/functions.R
    Modified:   code/genes.R
    Modified:   output/figures/feature-plot_ageing-pick-mrks-whole-integrated_.pdf
    Modified:   output/tables/novoseq-ageing_astr_recon-fix-mrk_logreg-sct.csv
    Modified:   output/tables/novoseq-ageing_astr_recon-fix-mrk_roc-sct.csv

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with
<code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<pre class="r"><code>combined_sct &lt;- LoadH5Seurat(here(data_dir, &quot;comb_astrocytes.bak.h5Seurat&quot;))</code></pre>
<pre class="r"><code>skim(combined_sct@meta.data)</code></pre>
<pre><code>── Data Summary ────────────────────────
                           Values                
Name                       combined_sct@meta.data
Number of rows             51734                 
Number of columns          78                    
_______________________                          
Column type frequency:                           
  character                38                    
  factor                   31                    
  numeric                  9                     
________________________                         
Group variables            None                  

── Variable type: character ────────────────────────────────────────────────────
   skim_variable        n_missing complete_rate min max empty n_unique
 1 Cell_ID                      0             1   2  45     0    50286
 2 Dataset                      0             1   2  16     0       19
 3 SRA_ID                       0             1   2  11     0       94
 4 Sample_ID                    0             1   2  31     0      104
 5 GEO_ID                       0             1   2  10     0       83
 6 Run10x                       0             1   2   9     0       11
 7 Technology                   0             1   2   7     0        5
 8 Strain                       0             1   2  14     0        9
 9 Diet                         0             1   2  13     0        6
10 Pooled                       0             1   2   3     0        4
11 Age                          0             1   2   9     0        4
12 Author_Region                0             1  12  41     0        9
13 inferred_sex                 0             1   1   2     0        4
14 Author_Exclude               0             1   2   3     0        3
15 Author_Class                 0             1   2  16     0       11
16 Author_CellType              0             1   2  47     0       71
17 Phase                        0             1   1   3     0        4
18 Batch_ID                     0             1  14  25     0       27
19 Author_Condition             0             1   2  34     0       24
20 Sex                          0             1   1   2     0        4
21 Author_Batch                 0             1   2   2     0        5
22 Author_Class_Curated         0             1   2  16     0       11
23 C2                           0             1   2   4     0        2
24 C7                           0             1   2   4     0        2
25 C25                          0             1   2   6     0        2
26 C66                          0             1   2   6     0        3
27 C185                         0             1   2   8     0       10
28 C286                         0             1   2   8     0       19
29 C465                         0             1   2   8     0       33
30 C2_named                     0             1   2  17     0        2
31 C7_named                     0             1   2  21     0        2
32 C25_named                    0             1   2  18     0        2
33 C66_named                    0             1   2  25     0        3
34 C185_named                   0             1   2  38     0       10
35 C286_named                   0             1   2  49     0       19
36 C465_named                   0             1   2  60     0       33
37 Region_predicted             0             1   2   2     0        1
38 Region_summarized            0             1   2   2     0        1
   whitespace
 1          0
 2          0
 3          0
 4          0
 5          0
 6          0
 7          0
 8          0
 9          0
10          0
11          0
12          0
13          0
14          0
15          0
16          0
17          0
18          0
19          0
20          0
21          0
22          0
23          0
24          0
25          0
26          0
27          0
28          0
29          0
30          0
31          0
32          0
33          0
34          0
35          0
36          0
37          0
38          0

── Variable type: factor ───────────────────────────────────────────────────────
   skim_variable                         n_missing complete_rate ordered
 1 integrated_snn_res.0.01                       0             1 FALSE  
 2 integrated_snn_res.0.0128607900386842         0             1 FALSE  
 3 integrated_snn_res.0.0159261774514492         0             1 FALSE  
 4 integrated_snn_res.0.019218924726995          0             1 FALSE  
 5 integrated_snn_res.0.022765300986261          0             1 FALSE  
 6 integrated_snn_res.0.0265957842901118         0             1 FALSE  
 7 integrated_snn_res.0.0307459397798481         0             1 FALSE  
 8 integrated_snn_res.0.0352575272462741         0             1 FALSE  
 9 integrated_snn_res.0.040179911243401          0             1 FALSE  
10 integrated_snn_res.0.0455718747630189         0             1 FALSE  
11 integrated_snn_res.0.0515039779635528         0             1 FALSE  
12 integrated_snn_res.0.0580616631325161         0             1 FALSE  
13 integrated_snn_res.0.0653493966487537         0             1 FALSE  
14 integrated_snn_res.0.0734962758235247         0             1 FALSE  
15 integrated_snn_res.0.0826637429070769         0             1 FALSE  
16 integrated_snn_res.0.0930563918997702         0             1 FALSE  
17 integrated_snn_res.0.104937418437845          0             1 FALSE  
18 integrated_snn_res.0.118651219606498          0             1 FALSE  
19 integrated_snn_res.0.13465732664678           0             1 FALSE  
20 integrated_snn_res.0.153582905371072          0             1 FALSE  
21 integrated_snn_res.0.176306867240956          0             1 FALSE  
22 integrated_snn_res.0.204100272968758          0             1 FALSE  
23 integrated_snn_res.0.238872500602171          0             1 FALSE  
24 integrated_snn_res.0.283629472972472          0             1 FALSE  
25 integrated_snn_res.0.34339279578797           0             1 FALSE  
26 integrated_snn_res.0.427229620759052          0             1 FALSE  
27 integrated_snn_res.0.553354125594305          0             1 FALSE  
28 integrated_snn_res.0.764552146711845          0             1 FALSE  
29 integrated_snn_res.1.19070222717945           0             1 FALSE  
30 integrated_snn_res.2.5                        0             1 FALSE  
31 seurat_clusters                               0             1 FALSE  
   n_unique top_counts                           
 1        4 1: 51727, 2: 3, 3: 2, 4: 2           
 2        4 1: 51727, 2: 3, 3: 2, 4: 2           
 3        4 1: 51727, 2: 3, 3: 2, 4: 2           
 4        4 1: 51727, 2: 3, 3: 2, 4: 2           
 5        4 1: 51727, 2: 3, 3: 2, 4: 2           
 6        4 1: 51727, 2: 3, 3: 2, 4: 2           
 7        4 1: 51727, 2: 3, 3: 2, 4: 2           
 8        4 1: 51727, 2: 3, 3: 2, 4: 2           
 9        4 1: 51727, 2: 3, 3: 2, 4: 2           
10        4 1: 51727, 2: 3, 3: 2, 4: 2           
11        4 1: 51727, 2: 3, 3: 2, 4: 2           
12        4 1: 51727, 2: 3, 3: 2, 4: 2           
13        4 1: 51727, 2: 3, 3: 2, 4: 2           
14        5 1: 51176, 2: 551, 3: 3, 4: 2         
15        5 1: 51171, 2: 556, 3: 3, 4: 2         
16        5 1: 44312, 2: 7415, 3: 3, 4: 2        
17        5 1: 41420, 2: 10307, 3: 3, 4: 2       
18        5 1: 44047, 2: 7680, 3: 3, 4: 2        
19        6 1: 37718, 2: 10497, 3: 3512, 4: 3    
20        6 1: 21686, 2: 19551, 3: 10490, 4: 3   
21        6 1: 21571, 2: 19552, 3: 10604, 4: 3   
22        7 1: 19522, 2: 17995, 3: 10659, 4: 3551
23        9 1: 19326, 2: 16224, 3: 9844, 4: 3554 
24        9 1: 19103, 2: 16431, 3: 9813, 4: 3577 
25        9 1: 19008, 2: 16385, 3: 9780, 4: 3609 
26       10 1: 12766, 2: 10476, 3: 9389, 4: 8879 
27       12 1: 10608, 2: 9988, 3: 8097, 4: 6133  
28       17 1: 6961, 2: 6764, 3: 6305, 4: 5333   
29       23 1: 6881, 2: 5347, 3: 5035, 4: 4241   
30       36 1: 3681, 2: 3413, 3: 3207, 4: 2545   
31       36 1: 3681, 2: 3413, 3: 3207, 4: 2545   

── Variable type: numeric ──────────────────────────────────────────────────────
  skim_variable            n_missing complete_rate       mean        sd       p0
1 nCount_RNA                       0         1     3196.      3467.     591     
2 nFeature_RNA                     0         1     1557.       867.     449     
3 nCount_SCT                       0         1     2920.      2060.     876     
4 nFeature_SCT                     0         1     1501.       733.     484     
5 percent_mt                       0         1        4.05       2.84     0     
6 percent_exclude_features      1449         0.972    0.195      0.0652   0.0117
7 S.Score                       1449         0.972   -0.00939    0.0430  -0.133 
8 G2M.Score                     1449         0.972   -0.0206     0.0555  -0.138 
9 k_tree                           0         1        5.20       3.29     1     
        p25       p50        p75      p100 hist 
1 1668      2340      3467       150220    ▇▁▁▁▁
2  980      1358      1851        11663    ▇▁▁▁▁
3 1811      2505      3278        16615    ▇▁▁▁▁
4  968      1342      1820         6664    ▇▃▁▁▁
5    1.59      3.90      6.11        15.0  ▇▇▅▁▁
6    0.144     0.197     0.236        2.54 ▇▁▁▁▁
7   -0.0392   -0.0166    0.0131       1.06 ▇▁▁▁▁
8   -0.0524   -0.0256    0.00210      1.76 ▇▁▁▁▁
9    2         5         7           17    ▇▅▃▁▁</code></pre>
<pre class="r"><code>combined_sct &lt;-
  Store_Palette_Seurat(
    seurat_object = combined_sct,
    palette = rev(brewer.pal(n = 11, name = &quot;Spectral&quot;)),
    palette_name = &quot;expr_Colour_Pal&quot;
  )</code></pre>
<p>Plot by nucleus</p>
<pre class="r"><code>plEmbCombBatch &lt;- DimPlot(combined_sct, reduction = &quot;umap&quot;,
                          group.by = &quot;Batch_ID&quot;,
                          label = TRUE, repel = TRUE) + NoLegend()
plEmbCombReg &lt;- DimPlot(combined_sct, reduction = &quot;umap&quot;,
                        group.by = &quot;Author_Region&quot;,
                        label = TRUE, repel = TRUE) + NoLegend()
plEmbCombBatch + plEmbCombReg</code></pre>
<p><img src="figure/plot-conv-integration-all-together-cloud.Rmd/pl-all-integrated-astrocytes-nucleus-1.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>p1 &lt;- DimPlot(combined_sct, label = T, repel = T) + ggtitle(&quot;Unsupervised clustering&quot;) + NoLegend()
p2 &lt;- DimPlot(combined_sct, label = T, repel = T, group.by = &quot;k_tree&quot;) + ggtitle(&quot;MRTree&quot;) + NoLegend()

p1 | p2</code></pre>
<p><img src="figure/plot-conv-integration-all-together-cloud.Rmd/pl-clustering-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>library(schex)
combined_sct &lt;- make_hexbin(combined_sct, nbins = 60, dimension_reduction = &quot;UMAP&quot;)
plot_hexbin_density(combined_sct)</code></pre>
<p><img src="figure/plot-conv-integration-all-together-cloud.Rmd/pl-schex-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot_hexbin_meta(combined_sct, col = &quot;nCount_RNA&quot;, action = &quot;median&quot;)</code></pre>
<p><img src="figure/plot-conv-integration-all-together-cloud.Rmd/pl-schex-2.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>combined_sct$k_tree &lt;- factor(combined_sct$k_tree)
plot_hexbin_meta(combined_sct, col = &quot;k_tree&quot;, action = &quot;majority&quot;)</code></pre>
<p><img src="figure/plot-conv-integration-all-together-cloud.Rmd/pl-schex-3.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>label_df &lt;- make_hexbin_label(combined_sct, col = &quot;k_tree&quot;)
pp &lt;- plot_hexbin_meta(combined_sct, col = &quot;k_tree&quot;, action = &quot;majority&quot;)
pp + ggrepel::geom_label_repel(data = label_df, aes(x = x, y = y, label = label), colour = &quot;black&quot;, 
    label.size = NA, fill = NA)</code></pre>
<p><img src="figure/plot-conv-integration-all-together-cloud.Rmd/pl-schex-k-tree-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>reg_df &lt;- make_hexbin_label(combined_sct, 
                            col = &quot;Author_Region&quot;)
pp2 &lt;- plot_hexbin_meta(combined_sct,
                        col = &quot;Author_Region&quot;,
                        action = &quot;majority&quot;)
pp2 + ggrepel::geom_label_repel(data = reg_df, 
                                aes(x = x, y = y,
                                    label = label), 
                                colour = &quot;black&quot;,
                                label.size = NA, fill = NA)</code></pre>
<p><img src="figure/plot-conv-integration-all-together-cloud.Rmd/pl-schex-regions-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plEmbCombCMrk &lt;- FeaturePlot_scCustom(
  combined_sct,
  colors_use = combined_sct@misc$expr_Colour_Pal,
  features = genes.embed[genes.embed %in%
                           rownames(combined_sct@assays$SCT@data)],
  max.cutoff = &quot;q99&quot;
)
plEmbCombCMrk</code></pre>
<p><img src="figure/plot-conv-integration-all-together-cloud.Rmd/pl-all-integrated-astrocytes-init-canon-markers-1.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>genes.embed[genes.embed %in% rownames(combined_sct@assays$SCT@scale.data)] |&gt;
  map(~ plot_density(combined_sct, .x))</code></pre>
<pre><code>[[1]]</code></pre>
<p><img src="figure/plot-conv-integration-all-together-cloud.Rmd/pl-all-integrated-astrocytes-init-canon-mrk-density-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
[[2]]</code></pre>
<p><img src="figure/plot-conv-integration-all-together-cloud.Rmd/pl-all-integrated-astrocytes-init-canon-mrk-density-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
[[3]]</code></pre>
<p><img src="figure/plot-conv-integration-all-together-cloud.Rmd/pl-all-integrated-astrocytes-init-canon-mrk-density-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
[[4]]</code></pre>
<p><img src="figure/plot-conv-integration-all-together-cloud.Rmd/pl-all-integrated-astrocytes-init-canon-mrk-density-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
[[5]]</code></pre>
<p><img src="figure/plot-conv-integration-all-together-cloud.Rmd/pl-all-integrated-astrocytes-init-canon-mrk-density-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
[[6]]</code></pre>
<p><img src="figure/plot-conv-integration-all-together-cloud.Rmd/pl-all-integrated-astrocytes-init-canon-mrk-density-6.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
[[7]]</code></pre>
<p><img src="figure/plot-conv-integration-all-together-cloud.Rmd/pl-all-integrated-astrocytes-init-canon-mrk-density-7.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
[[8]]</code></pre>
<p><img src="figure/plot-conv-integration-all-together-cloud.Rmd/pl-all-integrated-astrocytes-init-canon-mrk-density-8.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
[[9]]</code></pre>
<p><img src="figure/plot-conv-integration-all-together-cloud.Rmd/pl-all-integrated-astrocytes-init-canon-mrk-density-9.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
[[10]]</code></pre>
<p><img src="figure/plot-conv-integration-all-together-cloud.Rmd/pl-all-integrated-astrocytes-init-canon-mrk-density-10.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
[[11]]</code></pre>
<p><img src="figure/plot-conv-integration-all-together-cloud.Rmd/pl-all-integrated-astrocytes-init-canon-mrk-density-11.png" width="672" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.2 (2022-10-31)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 22.04.1 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3
LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] schex_1.12.0                shiny_1.7.4                
 [3] SingleCellExperiment_1.20.0 SummarizedExperiment_1.28.0
 [5] Biobase_2.58.0              GenomicRanges_1.50.2       
 [7] GenomeInfoDb_1.34.9         IRanges_2.32.0             
 [9] S4Vectors_0.36.1            BiocGenerics_0.44.0        
[11] MatrixGenerics_1.10.0       matrixStats_0.63.0         
[13] mrtree_0.0.0.9000           RColorBrewer_1.1-3         
[15] scCustomize_1.1.1           Nebulosa_1.8.0             
[17] swne_0.6.20                 patchwork_1.1.2.9000       
[19] UpSetR_1.4.0                glmGamPoi_1.10.2           
[21] sctransform_0.3.5           SeuratDisk_0.0.0.9020      
[23] SeuratWrappers_0.3.1        SeuratObject_4.1.3         
[25] Seurat_4.3.0                kableExtra_1.3.4           
[27] future_1.31.0               skimr_2.1.5                
[29] magrittr_2.0.3              lubridate_1.9.0            
[31] timechange_0.1.1            forcats_0.5.2              
[33] stringr_1.5.0               dplyr_1.1.0                
[35] purrr_1.0.1                 readr_2.1.3                
[37] tidyr_1.3.0                 tibble_3.1.8               
[39] ggplot2_3.4.1               tidyverse_1.3.2.9000       
[41] here_1.0.1                 

loaded via a namespace (and not attached):
  [1] rsvd_1.0.5              ica_1.0-3               svglite_2.1.0          
  [4] lmtest_0.9-40           rprojroot_2.0.3         crayon_1.5.2           
  [7] MASS_7.3-58.1           nlme_3.1-161            backports_1.4.1        
 [10] rlang_1.0.6             XVector_0.38.0          ROCR_1.0-11            
 [13] irlba_2.3.5.1           data.tree_1.0.0         bit64_4.0.5            
 [16] glue_1.6.2              parallel_4.2.2          vipor_0.4.5            
 [19] spatstat.sparse_3.0-0   spatstat.geom_3.0-6     tidyselect_1.2.0       
 [22] liger_2.0.1             fitdistrplus_1.1-8      zoo_1.8-11             
 [25] xtable_1.8-4            evaluate_0.20           cli_3.6.0              
 [28] zlibbioc_1.44.0         rstudioapi_0.14         miniUI_0.1.1.1         
 [31] sp_1.6-0                bslib_0.4.2             fastmatch_1.1-3        
 [34] treeio_1.23.0           maps_3.4.1              xfun_0.37              
 [37] askpass_1.1             usedist_0.4.0           cluster_2.1.4          
 [40] tidygraph_1.2.2         clusterGeneration_1.3.7 expm_0.999-6           
 [43] SymSim_0.0.0.9000       ggrepel_0.9.2.9999      ape_5.6-2              
 [46] listenv_0.9.0           dendextend_1.16.0       png_0.1-8              
 [49] withr_2.5.0             bitops_1.0-7            ggforce_0.4.1.9000     
 [52] plyr_1.8.8              pracma_2.4.2            coda_0.19-4            
 [55] pillar_1.8.1            GlobalOptions_0.1.2     cachem_1.0.6           
 [58] fs_1.6.1                scatterplot3d_0.3-42    hdf5r_1.3.7            
 [61] paletteer_1.5.0         vctrs_0.5.2             ellipsis_0.3.2         
 [64] generics_0.1.3          tools_4.2.2             beeswarm_0.4.0         
 [67] entropy_1.3.1           munsell_0.5.0           tweenr_2.0.2           
 [70] proxy_0.4-27            DelayedArray_0.24.0     fastmap_1.1.0          
 [73] compiler_4.2.2          abind_1.4-5             httpuv_1.6.9           
 [76] ggimage_0.3.1           plotly_4.10.1           GenomeInfoDbData_1.2.9 
 [79] gridExtra_2.3           workflowr_1.7.0         lattice_0.20-45        
 [82] deldir_1.0-6            snow_0.4-4              utf8_1.2.3             
 [85] later_1.3.0             jsonlite_1.8.4          concaveman_1.1.0       
 [88] scales_1.2.1            tidytree_0.4.2          pbapply_1.7-0          
 [91] lazyeval_0.2.2          promises_1.2.0.1        R.utils_2.12.2         
 [94] goftest_1.2-3           spatstat.utils_3.0-1    reticulate_1.28        
 [97] checkmate_2.1.0         rmarkdown_2.20          cowplot_1.1.1          
[100] webshot_0.5.4           Rtsne_0.16              uwot_0.1.14            
[103] igraph_1.3.5            survival_3.4-0          numDeriv_2016.8-1.1    
[106] yaml_2.3.7              plotrix_3.8-2           systemfonts_1.0.4      
[109] htmltools_0.5.4         graphlayouts_0.8.4      quadprog_1.5-8         
[112] viridisLite_0.4.1       digest_0.6.31           mime_0.12              
[115] repr_1.1.4              yulab.utils_0.0.6       future.apply_1.10.0    
[118] ggmin_0.0.0.9000        remotes_2.4.2           data.table_1.14.8      
[121] R.oo_1.25.0             splines_4.2.2           labeling_0.4.2         
[124] rematch2_2.1.2          RCurl_1.98-1.9          ks_1.14.0              
[127] hms_1.1.2               colorspace_2.1-0        base64enc_0.1-3        
[130] BiocManager_1.30.19     mnormt_2.1.1            ggbeeswarm_0.7.1.9000  
[133] shape_1.4.6             aplot_0.1.9             ggrastr_1.0.1          
[136] sass_0.4.5              Rcpp_1.0.10             mclust_6.0.0           
[139] RANN_2.6.1              mvtnorm_1.1-3           circlize_0.4.15        
[142] NNLM_0.4.4              fansi_1.0.4             tzdb_0.3.0             
[145] parallelly_1.34.0       R6_2.5.1                grid_4.2.2             
[148] ggridges_0.5.4          lifecycle_1.0.3         phytools_1.2-0         
[151] leiden_0.4.3            phangorn_2.10.0         jquerylib_0.1.4        
[154] snakecase_0.11.0        Matrix_1.5-3            RcppAnnoy_0.0.20       
[157] spatstat.explore_3.0-6  htmlwidgets_1.6.1       umap_0.2.9.0           
[160] polyclip_1.10-4         gridGraphics_0.5-1      optimParallel_1.0-2    
[163] rvest_1.0.3             mgcv_1.8-41             globals_0.16.2         
[166] openssl_2.0.5           spatstat.random_3.1-3   progressr_0.13.0       
[169] codetools_0.2-18        FNN_1.1.3.1             RSpectra_0.16-1        
[172] R.methodsS3_1.8.2       gtable_0.3.1            git2r_0.30.1           
[175] ggfun_0.0.9             tensor_1.5              httr_1.4.4             
[178] highr_0.10              KernSmooth_2.23-20      stringi_1.7.12         
[181] vroom_1.6.0             reshape2_1.4.4          farver_2.1.1           
[184] viridis_0.6.2           hexbin_1.28.2           magick_2.7.3           
[187] ggtree_3.7.1.002        xml2_1.3.3              combinat_0.0-8         
[190] ggplotify_0.1.0         scattermore_0.8         bit_4.0.5              
[193] clustree_0.5.0          spatstat.data_3.0-0     ggraph_2.1.0.9000      
[196] janitor_2.2.0.9000      pkgconfig_2.0.3         ggprism_1.0.4          
[199] knitr_1.42             </code></pre>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
