---
title: "Differential expression analysis of individual hypothalamic nuclei astrocytes evaluation dataset"
author: "Evgenii O. Tretiakov"
date: "`r Sys.Date()`"
format:
  html:
    toc: true
    df-print: paged
    code-fold: true
    fig-width: 9
    fig-height: 12
    fig-format: retina
    fig-responsive: true
    fig-dpi: 600
execute:
  keep-md: false
  echo: true
  error: false
  message: false
  warning: false
  debug: false
knitr:
  opts_chunk:
    autodep: true
    fig.align: center
    fig.retina: 2
    fig.width: 14
    fig.height: 12
---

```{r setup, include = FALSE}
DOCNAME <- "de-astrocytes-between-regions-on-evaluation-datasets"
NOW <- Sys.time()

# Time chunks during knitting
knitr::knit_hooks$set(timeit = function(before) {
  if (before) {
    print(paste("Start:", Sys.time()))
    NOW <<- Sys.time()
  } else {
    print(paste("Stop:", Sys.time()))
    print(Sys.time() - NOW)
  }
})

knitr::knit_hooks$set(debug = function(before, options, envir) {
  if (!before) {
    message(
      paste(names(envir), as.list(envir),
        sep = " = ", collapse = "\n"
      )
    )
  }
})

knitr::opts_chunk$set(
  cache          = FALSE,
  dev            = c("png", "pdf"),
  timeit         = TRUE
)
```

## Load data and setup parameters

```{r libraries, cache=FALSE}
# Load tidyverse infrastructure packages
suppressPackageStartupMessages({
  library(future)
  library(here)
  library(tidyverse)
  library(magrittr)
  library(stringr)
  library(skimr)
  library(RColorBrewer)
  library(viridis)
})


# Load packages for scRNA-seq analysis and visualisation
suppressPackageStartupMessages({
  library(ggplot2)
  library(cowplot)
  library(patchwork)
  library(ggstatsplot)
  library(anndata)
  library(sceasy)
  library(Seurat)
  library(SeuratDisk)
  library(SeuratWrappers)
  library(scCustomize)
})

sc <- import("scanpy", convert = FALSE)
```

### Set paths

```{r paths}
src_dir <- here("code")
data_dir <- here("data")
output_dir <- here("output")
plots_dir <- here(output_dir, "figures")
tables_dir <- here(output_dir, "tables")
```

### Load helper functions and gene-sets

```{r source, cache = FALSE}
source(here(src_dir, "genes.R"))
source(here(src_dir, "functions.R"))
```

### Set fixed variables

```{r params-computation, cache = FALSE}
# set seed
reseed <- 42
set.seed(seed = reseed)

# Parameters for parallel execution
n_cores <- 8
plan("multisession", workers = n_cores)
options(
  future.globals.maxSize = 1999999 * 1024^2,
  future.rng.onMisuse = "ignore"
)
plan()


# ggplot2 theme
theme_set(ggmin::theme_powerpoint())
```

```{r params}
bioproject <- "individual_hypothalamic_nuclei_astrocytes_evaluation_dataset"
cb_fpr <- 0.001
connectivity_model <- "min_tree"
k <- 10
metric <- "euclidean"
signature <- 100
```

## Load predicted astrocytes data and subset from Hajdarovic KH et al 2022

```{r load-data, cache = FALSE}
anndata <- sc$read(here(
  data_dir,
  sprintf("data/data/resolved_subregions_by_microclusters-replic-3/replic2-best-subregional_supervised-evaluation-astrocytes_datasets-msp_%s-metric_%s-k_%s-sign_%s-amb_%s.h5ad", connectivity_model, metric, k, signature, cb_fpr)
))
```

### Convert adata object to R AnnDataR6 object.
```{r convert-to-r}
adata <- py_to_r(anndata)
class(adata)
class(adata$X)
adata
```

```{r convert-to-seurat}
srt_path <- here(
  data_dir,
  sprintf("best_xgboost-subregional-astrocytes_dataset-msp_%s-metric_%s-k_%s-sign_%s-amb_%s.h5Seurat", connectivity_model, metric, k, signature, cb_fpr)
)

expr_mtx <- t(as.matrix(adata$raw$X))
colnames(expr_mtx) <- rownames(adata$X)
rownames(expr_mtx) <- adata$var_names
srt <- CreateSeuratObject(
  expr_mtx,
  assay = "RNA",
  project = "individual_hypothalamic_nuclei_astrocytes_evaluation_dataset",
  meta.data = as.data.frame(adata$obs),
  row.names = colnames(adata$X)
)


densmap_logit <- adata$obsm$X_umap_logit
colnames(densmap_logit) <- c("densMAP_logit_1", "densMAP_logit_2")
srt[["densmap_logit"]] <- CreateDimReducObject(embeddings = densmap_logit, key = "densMAP_logit_", assay = DefaultAssay(srt))
# srt <- ProjectDim(srt, reduction = "densmap_logit")

densmap_xgboost <- adata$obsm$X_umap
colnames(densmap_xgboost) <- c("densMAP_xgboost_1", "densMAP_xgboost_2")
srt[["densmap_xgboost"]] <- CreateDimReducObject(embeddings = densmap_xgboost, key = "densMAP_xgboost_", assay = DefaultAssay(srt))
# srt <- ProjectDim(srt, reduction = "densmap_xgboost")

Idents(srt) <- "predictions_xgboost_eval"

srt <- Store_Palette_Seurat(seurat_object = srt, palette = rev(brewer.pal(n = 11, name = "Spectral")), palette_name = "expr_Colour_Pal")

SaveH5Seurat(srt, filename = srt_path, overwrite = TRUE)
```

```{r load-seurat}
srt <- LoadH5Seurat(file = srt_path)
print(srt)
rm(adata, anndata, expr_mtx, densmap_logit, densmap_xgboost)
invisible(gc())
```


```{r scale-data}
srt <- NormalizeData(srt)
srt <- FindVariableFeatures(srt, selection.method = "vst", nfeatures = 5000)
all.genes <- rownames(srt)
srt <- ScaleData(srt, features = all.genes)
# srt <- ScaleData(srt, features = union(VariableFeatures(srt), gene_int[gene_int %in% all.genes]))
```

```{r origin-find-marker-genes}
invisible(gc())
plan("sequential")
Idents(srt) <- "codes_eval"
invisible(gc())
plan("multisession", workers = n_cores)
all_markers_genes_origin <-
  FindAllMarkers(object = srt, verbose = T, test.use = "LR", latent.vars = c("log10GenesPerUMI", "percent_mito_ribo"), only.pos = T, min.pct = 0.005, logfc.threshold = 0.1, max.cells.per.ident = 2000, random.seed = reseed)
all_markers_genes_origin <-
  all_markers_genes_origin %>%
  Add_Pct_Diff()
write_csv(all_markers_genes_origin, here(data_dir, sprintf("%s-all-marker-genes-origin.csv", "individual_hypothalamic_nuclei_astrocytes_evaluation_dataset")))
all_markers_genes_origin
```

```{r astrotrap-find-marker-genes}
invisible(gc())
plan("sequential")
Idents(srt) <- "predictions_xgboost_eval"
invisible(gc())
plan("multisession", workers = n_cores)
all_markers_genes <-
  FindAllMarkers(object = srt, verbose = T, test.use = "LR", latent.vars = c("log10GenesPerUMI", "percent_mito_ribo"), only.pos = T, min.pct = 0.005, logfc.threshold = 0.1, max.cells.per.ident = 2000, random.seed = reseed)
all_markers_genes <-
  all_markers_genes %>%
  Add_Pct_Diff()
write_csv(all_markers_genes, here(data_dir, sprintf("%s-all-marker-genes-astrotrap.csv", "individual_hypothalamic_nuclei_astrocytes_evaluation_dataset")))
all_markers_genes
```


```{r extract-top-30-genes-astrotrap}
all_markers_genes_astrotrap <- all_markers_genes
all_markers_genes <- all_markers_genes %>%
  filter(pct.2 < 0.3, pct_diff > 0.1) %>%
  group_by(cluster) %>%
  arrange(desc(avg_log2FC)) %>%
  arrange(cluster)
all_markers_genes2 <- all_markers_genes %>%
  filter(gene %in% c(npr, nmr, transcription_factors)) %>%
  group_by(cluster) %>%
  arrange(desc(avg_log2FC)) %>%
  arrange(cluster)

at_top_5 <- Extract_Top_Markers(marker_dataframe = all_markers_genes, num_genes = 5, group_by = "cluster", gene_column = "gene", rank_by = "avg_log2FC", make_unique = TRUE, named_vector = FALSE)
at_top_30 <- Extract_Top_Markers(marker_dataframe = all_markers_genes, num_genes = 30, group_by = "cluster", gene_column = "gene", rank_by = "avg_log2FC")
at_top_10s <- Extract_Top_Markers(marker_dataframe = all_markers_genes2, num_genes = 10, group_by = "cluster", gene_column = "gene", rank_by = "avg_log2FC", make_unique = TRUE, named_vector = FALSE)
at_top_30s <- Extract_Top_Markers(marker_dataframe = all_markers_genes2, num_genes = 30, group_by = "cluster", gene_column = "gene", rank_by = "avg_log2FC")
```

```{r extract-top-30-genes-origin}
all_markers_genes_origin <- all_markers_genes_origin %>%
  filter(pct.2 < 0.3, pct_diff > 0.1) %>%
  group_by(cluster) %>%
  arrange(desc(avg_log2FC)) %>%
  arrange(cluster)
all_markers_genes_origin2 <- all_markers_genes_origin %>%
  filter(gene %in% c(npr, nmr, transcription_factors)) %>%
  group_by(cluster) %>%
  arrange(desc(avg_log2FC)) %>%
  arrange(cluster)

o_top_5 <- Extract_Top_Markers(marker_dataframe = all_markers_genes_origin, num_genes = 5, group_by = "cluster", gene_column = "gene", rank_by = "avg_log2FC", make_unique = TRUE, named_vector = FALSE)
o_top_30 <- Extract_Top_Markers(marker_dataframe = all_markers_genes_origin, num_genes = 30, group_by = "cluster", gene_column = "gene", rank_by = "avg_log2FC")
o_top_10s <- Extract_Top_Markers(marker_dataframe = all_markers_genes_origin2, num_genes = 10, group_by = "cluster", gene_column = "gene", rank_by = "avg_log2FC", make_unique = TRUE, named_vector = FALSE)
o_top_30s <- Extract_Top_Markers(marker_dataframe = all_markers_genes_origin2, num_genes = 30, group_by = "cluster", gene_column = "gene", rank_by = "avg_log2FC")
```

```{r save-markers-featureplots-at_genes}
Iterate_FeaturePlot_scCustom(
  reduction = "densmap_xgboost",
  na_cutoff = NA,
  pt.size = 4,
  order = TRUE,
  alpha_na_exp = 0.1,
  alpha_exp = 0.75, seurat_object = srt, gene_list = at_top_30, single_pdf = FALSE, file_path = plots_dir, file_type = ".pdf", file_name = sprintf("%s_at.pdf", bioproject), colors_use = srt@misc$expr_Colour_Pal
)
```

```{r save-markers-featureplots-at_sgenes}
Iterate_FeaturePlot_scCustom(
  reduction = "densmap_xgboost",
  na_cutoff = NA,
  pt.size = 4,
  order = TRUE,
  alpha_na_exp = 0.1,
  alpha_exp = 0.75, seurat_object = srt, gene_list = at_top_30s, single_pdf = FALSE, file_path = plots_dir, file_type = ".pdf", file_name = sprintf("%s_ats.pdf", bioproject), colors_use = srt@misc$expr_Colour_Pal
)
```

```{r save-markers-featureplots-o_genes}
Iterate_FeaturePlot_scCustom(
  reduction = "densmap_xgboost",
  na_cutoff = NA,
  pt.size = 4,
  order = TRUE,
  alpha_na_exp = 0.1,
  alpha_exp = 0.75, seurat_object = srt, gene_list = o_top_30, single_pdf = FALSE, file_path = plots_dir, file_type = ".pdf", file_name = sprintf("%s_o.pdf", bioproject), colors_use = srt@misc$expr_Colour_Pal
)
```

```{r save-markers-featureplots-o_sgenes}
Iterate_FeaturePlot_scCustom(
  reduction = "densmap_xgboost",
  na_cutoff = NA,
  pt.size = 4,
  order = TRUE,
  alpha_na_exp = 0.1,
  alpha_exp = 0.75, seurat_object = srt, gene_list = o_top_30s, single_pdf = FALSE, file_path = plots_dir, file_type = ".pdf", file_name = sprintf("%s_os.pdf", bioproject), colors_use = srt@misc$expr_Colour_Pal
)
```

```{r plt-dotplot-dendrogram-at_genes, fig.width=6, fig.height=16}
Clustered_DotPlot(seurat_object = srt, colors_use_exp = viridis(n = 30, alpha = .75, direction = -1, option = "E"), features = at_top_5, k = 9)
```

```{r plt-dotplot-dendrogram-at_sgenes, fig.width=6, fig.height=16}
Clustered_DotPlot(seurat_object = srt, colors_use_exp = viridis(n = 30, alpha = .75, direction = -1, option = "E"), features = at_top_10s, k = 9)
```

```{r plt-dotplot-dendrogram-at_genes-npr, fig.width=6, fig.height=18}
DotPlot_scCustom(seurat_object = srt, colors_use = viridis(n = 30, alpha = .75, direction = -1, option = "E"), features = npr[npr %in% rownames(GetAssayData(object = srt, slot = "scale.data"))], flip_axes = T, x_lab_rotate = TRUE)
```

```{r plt-dotplot-dendrogram-at_genes-nmr, fig.width=6, fig.height=24}
DotPlot_scCustom(seurat_object = srt, colors_use = viridis(n = 30, alpha = .75, direction = -1, option = "E"), features = nmr[nmr %in% rownames(GetAssayData(object = srt, slot = "scale.data"))], flip_axes = T, x_lab_rotate = TRUE)
```

```{r plt-dotplot-dendrogram-at_genes-adgen, fig.width=6, fig.height=24}
DotPlot_scCustom(seurat_object = srt, colors_use = viridis(n = 30, alpha = .75, direction = -1, option = "E"), features = genes.embed[genes.embed %in% rownames(GetAssayData(object = srt, slot = "scale.data"))], flip_axes = T, x_lab_rotate = TRUE)
```

```{r plt-dotplot-dendrogram-at_genes-mitochondrial, fig.width=6, fig.height=24}
DotPlot_scCustom(seurat_object = srt, colors_use = viridis(n = 30, alpha = .75, direction = -1, option = "E"), features = mitochondrial[mitochondrial %in% rownames(GetAssayData(object = srt, slot = "scale.data"))], flip_axes = T, x_lab_rotate = TRUE)
```

```{r at_heatmap, fig.width=18, fig.height=18}
DoHeatmap(srt, group.by = "predictions_xgboost_eval", group.colors = c("0" = "black", "1" = "purple", "2" = "blue", "3" = "darkgreen", "4" = "green", "5" = "yellow", "6" = "red", "7" = "grey"), features = at_top_10s, size = 4, angle = 0) + NoLegend()
```


```{r at_de-pairs}
de_01 <- FindMarkers(srt,
  assay = "RNA", ident.1 = "0", ident.2 = "1",
  test.use = "LR",
  only.pos = F,
  min.pct = 0.01
) %>% rownames_to_column("gene")

de_02 <- FindMarkers(srt,
  assay = "RNA", ident.1 = "0", ident.2 = "2",
  test.use = "LR",
  only.pos = F,
  min.pct = 0.01
) %>% rownames_to_column("gene")

de_03 <- FindMarkers(srt,
  assay = "RNA", ident.1 = "0", ident.2 = "3",
  test.use = "LR",
  only.pos = F,
  min.pct = 0.01
) %>% rownames_to_column("gene")

de_04 <- FindMarkers(srt,
  assay = "RNA", ident.1 = "0", ident.2 = "4",
  test.use = "LR",
  only.pos = F,
  min.pct = 0.01
) %>% rownames_to_column("gene")

de_05 <- FindMarkers(srt,
  assay = "RNA", ident.1 = "0", ident.2 = "5",
  test.use = "LR",
  only.pos = F,
  min.pct = 0.01
) %>% rownames_to_column("gene")

de_06 <- FindMarkers(srt,
  assay = "RNA", ident.1 = "0", ident.2 = "6",
  test.use = "LR",
  only.pos = F,
  min.pct = 0.01
) %>% rownames_to_column("gene")

de_07 <- FindMarkers(srt,
  assay = "RNA", ident.1 = "0", ident.2 = "7",
  test.use = "LR",
  only.pos = F,
  min.pct = 0.01
) %>% rownames_to_column("gene")

de_12 <- FindMarkers(srt,
  assay = "RNA", ident.1 = "1", ident.2 = "2",
  test.use = "LR",
  only.pos = F,
  min.pct = 0.01
) %>% rownames_to_column("gene")

de_13 <- FindMarkers(srt,
  assay = "RNA", ident.1 = "1", ident.2 = "3",
  test.use = "LR",
  only.pos = F,
  min.pct = 0.01
) %>% rownames_to_column("gene")

de_14 <- FindMarkers(srt,
  assay = "RNA", ident.1 = "1", ident.2 = "4",
  test.use = "LR",
  only.pos = F,
  min.pct = 0.01
) %>% rownames_to_column("gene")

de_15 <- FindMarkers(srt,
  assay = "RNA", ident.1 = "1", ident.2 = "5",
  test.use = "LR",
  only.pos = F,
  min.pct = 0.01
) %>% rownames_to_column("gene")

de_16 <- FindMarkers(srt,
  assay = "RNA", ident.1 = "1", ident.2 = "6",
  test.use = "LR",
  only.pos = F,
  min.pct = 0.01
) %>% rownames_to_column("gene")

de_17 <- FindMarkers(srt,
  assay = "RNA", ident.1 = "1", ident.2 = "7",
  test.use = "LR",
  only.pos = F,
  min.pct = 0.01
) %>% rownames_to_column("gene")

de_23 <- FindMarkers(srt,
  assay = "RNA", ident.1 = "2", ident.2 = "3",
  test.use = "LR",
  only.pos = F,
  min.pct = 0.01
) %>% rownames_to_column("gene")

de_24 <- FindMarkers(srt,
  assay = "RNA", ident.1 = "2", ident.2 = "4",
  test.use = "LR",
  only.pos = F,
  min.pct = 0.01
) %>% rownames_to_column("gene")

de_25 <- FindMarkers(srt,
  assay = "RNA", ident.1 = "2", ident.2 = "5",
  test.use = "LR",
  only.pos = F,
  min.pct = 0.01
) %>% rownames_to_column("gene")

de_26 <- FindMarkers(srt,
  assay = "RNA", ident.1 = "2", ident.2 = "6",
  test.use = "LR",
  only.pos = F,
  min.pct = 0.01
) %>% rownames_to_column("gene")

de_27 <- FindMarkers(srt,
  assay = "RNA", ident.1 = "2", ident.2 = "7",
  test.use = "LR",
  only.pos = F,
  min.pct = 0.01
) %>% rownames_to_column("gene")

de_34 <- FindMarkers(srt,
  assay = "RNA", ident.1 = "3", ident.2 = "4",
  test.use = "LR",
  only.pos = F,
  min.pct = 0.01
) %>% rownames_to_column("gene")

de_35 <- FindMarkers(srt,
  assay = "RNA", ident.1 = "3", ident.2 = "5",
  test.use = "LR",
  only.pos = F,
  min.pct = 0.01
) %>% rownames_to_column("gene")

de_36 <- FindMarkers(srt,
  assay = "RNA", ident.1 = "3", ident.2 = "6",
  test.use = "LR",
  only.pos = F,
  min.pct = 0.01
) %>% rownames_to_column("gene")

de_37 <- FindMarkers(srt,
  assay = "RNA", ident.1 = "3", ident.2 = "7",
  test.use = "LR",
  only.pos = F,
  min.pct = 0.01
) %>% rownames_to_column("gene")

de_45 <- FindMarkers(srt,
  assay = "RNA", ident.1 = "4", ident.2 = "5",
  test.use = "LR",
  only.pos = F,
  min.pct = 0.01
) %>% rownames_to_column("gene")

de_46 <- FindMarkers(srt,
  assay = "RNA", ident.1 = "4", ident.2 = "6",
  test.use = "LR",
  only.pos = F,
  min.pct = 0.01
) %>% rownames_to_column("gene")

de_47 <- FindMarkers(srt,
  assay = "RNA", ident.1 = "4", ident.2 = "7",
  test.use = "LR",
  only.pos = F,
  min.pct = 0.01
) %>% rownames_to_column("gene")

de_56 <- FindMarkers(srt,
  assay = "RNA", ident.1 = "5", ident.2 = "6",
  test.use = "LR",
  only.pos = F,
  min.pct = 0.01
) %>% rownames_to_column("gene")

de_57 <- FindMarkers(srt,
  assay = "RNA", ident.1 = "5", ident.2 = "7",
  test.use = "LR",
  only.pos = F,
  min.pct = 0.01
) %>% rownames_to_column("gene")

de_67 <- FindMarkers(srt,
  assay = "RNA", ident.1 = "6", ident.2 = "7",
  test.use = "LR",
  only.pos = F,
  min.pct = 0.01
) %>% rownames_to_column("gene")


at_top_genes <- list()

combinations <- list(
  contrast01 = de_01, contrast02 = de_02, contrast03 = de_03,
  contrast04 = de_04, contrast05 = de_05, contrast06 = de_06,
  contrast07 = de_07, contrast12 = de_12, contrast13 = de_13,
  contrast14 = de_14, contrast15 = de_15, contrast16 = de_16,
  contrast17 = de_17, contrast23 = de_23, contrast24 = de_24,
  contrast25 = de_25, contrast26 = de_26, contrast27 = de_27,
  contrast34 = de_34, contrast35 = de_35, contrast36 = de_36,
  contrast37 = de_37, contrast45 = de_45, contrast46 = de_46,
  contrast47 = de_47, contrast56 = de_56, contrast57 = de_57,
  contrast67 = de_67
) %>% compact()

# Iterate through the 'combinations' list
for (contrast_name in names(combinations)) {
  # Create a unique file name for each data frame
  file_name <- paste0(contrast_name, "_result_astrotrap.csv")

  # Save the data frame as a CSV file
  write_csv(combinations[[contrast_name]], here(tables_dir, file_name))

  # Print a message to indicate progress
  print(paste0("Saved: ", file_name))
}

for (contrast in names(combinations)) {
  top_upregulated <- combinations[[contrast]] %>%
    filter(p_val_adj < 0.05, avg_log2FC > 0, gene %in% gene_int) %>%
    arrange(desc(avg_log2FC)) %>%
    slice_head(n = 5)
  top_downregulated <- combinations[[contrast]] %>%
    filter(p_val_adj < 0.05, avg_log2FC < 0, gene %in% gene_int) %>%
    arrange(avg_log2FC) %>%
    slice_head(n = 5)

  at_top_genes[[contrast]] <- c(top_upregulated$gene, top_downregulated$gene)
}
```

```{r at_collapse-clusters}
DefaultAssay(srt) <- "RNA"
at_cluster.averages <- AverageExpression(srt, return.seurat = TRUE)
DefaultAssay(at_cluster.averages) <- "RNA"

at_cluster.averages
```

```{r at_de-pairs-plot, fig.width=14, fig.height=13}
plot_pairs <- function(highlight_genes, pair_name) {
  i <- pair_name %>% str_sub(9, 9)
  j <- pair_name %>% str_sub(10, 10)
  print(paste0("Comparison: ", i, " vs ", j))
  plot <- CellScatter(at_cluster.averages, cell1 = i, cell2 = j, features = gene_int[gene_int %in% rownames(at_cluster.averages)], highlight = highlight_genes)
  return(plot)
}
plots <- at_top_genes %>% imap(plot_pairs)
at_plt_grid <- cowplot::plot_grid(plotlist = plots, ncol = 7, labels = names(at_top_genes) %>% map_chr(str_sub, 9, 10), label_size = 8)
at_plt_grid

save_plot(here(plots_dir, "de-pairs-plot_astrotrap.pdf"), at_plt_grid, base_width = 6, base_height = 7, base_asp = 1, ncol = 7, limitsize = FALSE)
```

```{r plt-dotplot-dendrogram-o_genes, fig.width=6, fig.height=16}
Idents(srt) <- "codes_eval"
Clustered_DotPlot(seurat_object = srt, colors_use_exp = viridis(n = 30, alpha = .75, direction = -1, option = "E"), features = o_top_5, k = 9)
```

```{r plt-dotplot-dendrogram-o_sgenes, fig.width=6, fig.height=16}
Clustered_DotPlot(seurat_object = srt, colors_use_exp = viridis(n = 30, alpha = .75, direction = -1, option = "E"), features = o_top_10s, k = 9)
```

```{r plt-dotplot-dendrogram-o_genes-npr, fig.width=6, fig.height=18}
DotPlot_scCustom(seurat_object = srt, colors_use = viridis(n = 30, alpha = .75, direction = -1, option = "E"), features = npr[npr %in% rownames(GetAssayData(object = srt, slot = "scale.data"))], flip_axes = T, x_lab_rotate = TRUE)
```

```{r plt-dotplot-dendrogram-o_genes-nmr, fig.width=6, fig.height=24}
DotPlot_scCustom(seurat_object = srt, colors_use = viridis(n = 30, alpha = .75, direction = -1, option = "E"), features = nmr[nmr %in% rownames(GetAssayData(object = srt, slot = "scale.data"))], flip_axes = T, x_lab_rotate = TRUE)
```

```{r plt-dotplot-dendrogram-o_genes-adgen, fig.width=6, fig.height=24}
DotPlot_scCustom(seurat_object = srt, colors_use = viridis(n = 30, alpha = .75, direction = -1, option = "E"), features = genes.embed[genes.embed %in% rownames(GetAssayData(object = srt, slot = "scale.data"))], flip_axes = T, x_lab_rotate = TRUE)
```

```{r plt-dotplot-dendrogram-o_genes-mitochondrial, fig.width=6, fig.height=24}
DotPlot_scCustom(seurat_object = srt, colors_use = viridis(n = 30, alpha = .75, direction = -1, option = "E"), features = mitochondrial[mitochondrial %in% rownames(GetAssayData(object = srt, slot = "scale.data"))], flip_axes = T, x_lab_rotate = TRUE)
```

```{r o_heatmap, fig.width=18, fig.height=18}
DoHeatmap(srt, group.by = "region", group.colors = c("ARC" = "black", "LHA" = "purple", "MnPO" = "blue", "POA" = "darkgreen", "PVN" = "green", "SCN" = "yellow", "VMH" = "red", "VPH" = "grey"), features = o_top_10s, size = 4, angle = 0) + NoLegend()
```



```{r o_de-pairs}
de_01 <- FindMarkers(srt,
  assay = "RNA", ident.1 = "0", ident.2 = "1",
  test.use = "LR",
  only.pos = F,
  min.pct = 0.01
) %>% rownames_to_column("gene")

de_02 <- FindMarkers(srt,
  assay = "RNA", ident.1 = "0", ident.2 = "2",
  test.use = "LR",
  only.pos = F,
  min.pct = 0.01
) %>% rownames_to_column("gene")

de_03 <- FindMarkers(srt,
  assay = "RNA", ident.1 = "0", ident.2 = "3",
  test.use = "LR",
  only.pos = F,
  min.pct = 0.01
) %>% rownames_to_column("gene")

de_04 <- FindMarkers(srt,
  assay = "RNA", ident.1 = "0", ident.2 = "4",
  test.use = "LR",
  only.pos = F,
  min.pct = 0.01
) %>% rownames_to_column("gene")

de_05 <- FindMarkers(srt,
  assay = "RNA", ident.1 = "0", ident.2 = "5",
  test.use = "LR",
  only.pos = F,
  min.pct = 0.01
) %>% rownames_to_column("gene")

de_06 <- FindMarkers(srt,
  assay = "RNA", ident.1 = "0", ident.2 = "6",
  test.use = "LR",
  only.pos = F,
  min.pct = 0.01
) %>% rownames_to_column("gene")

de_07 <- FindMarkers(srt,
  assay = "RNA", ident.1 = "0", ident.2 = "7",
  test.use = "LR",
  only.pos = F,
  min.pct = 0.01
) %>% rownames_to_column("gene")

de_12 <- FindMarkers(srt,
  assay = "RNA", ident.1 = "1", ident.2 = "2",
  test.use = "LR",
  only.pos = F,
  min.pct = 0.01
) %>% rownames_to_column("gene")

de_13 <- FindMarkers(srt,
  assay = "RNA", ident.1 = "1", ident.2 = "3",
  test.use = "LR",
  only.pos = F,
  min.pct = 0.01
) %>% rownames_to_column("gene")

de_14 <- FindMarkers(srt,
  assay = "RNA", ident.1 = "1", ident.2 = "4",
  test.use = "LR",
  only.pos = F,
  min.pct = 0.01
) %>% rownames_to_column("gene")

de_15 <- FindMarkers(srt,
  assay = "RNA", ident.1 = "1", ident.2 = "5",
  test.use = "LR",
  only.pos = F,
  min.pct = 0.01
) %>% rownames_to_column("gene")

de_16 <- FindMarkers(srt,
  assay = "RNA", ident.1 = "1", ident.2 = "6",
  test.use = "LR",
  only.pos = F,
  min.pct = 0.01
) %>% rownames_to_column("gene")

de_17 <- FindMarkers(srt,
  assay = "RNA", ident.1 = "1", ident.2 = "7",
  test.use = "LR",
  only.pos = F,
  min.pct = 0.01
) %>% rownames_to_column("gene")

de_23 <- FindMarkers(srt,
  assay = "RNA", ident.1 = "2", ident.2 = "3",
  test.use = "LR",
  only.pos = F,
  min.pct = 0.01
) %>% rownames_to_column("gene")

de_24 <- FindMarkers(srt,
  assay = "RNA", ident.1 = "2", ident.2 = "4",
  test.use = "LR",
  only.pos = F,
  min.pct = 0.01
) %>% rownames_to_column("gene")

de_25 <- FindMarkers(srt,
  assay = "RNA", ident.1 = "2", ident.2 = "5",
  test.use = "LR",
  only.pos = F,
  min.pct = 0.01
) %>% rownames_to_column("gene")

de_26 <- FindMarkers(srt,
  assay = "RNA", ident.1 = "2", ident.2 = "6",
  test.use = "LR",
  only.pos = F,
  min.pct = 0.01
) %>% rownames_to_column("gene")

de_27 <- FindMarkers(srt,
  assay = "RNA", ident.1 = "2", ident.2 = "7",
  test.use = "LR",
  only.pos = F,
  min.pct = 0.01
) %>% rownames_to_column("gene")

de_34 <- FindMarkers(srt,
  assay = "RNA", ident.1 = "3", ident.2 = "4",
  test.use = "LR",
  only.pos = F,
  min.pct = 0.01
) %>% rownames_to_column("gene")

de_35 <- FindMarkers(srt,
  assay = "RNA", ident.1 = "3", ident.2 = "5",
  test.use = "LR",
  only.pos = F,
  min.pct = 0.01
) %>% rownames_to_column("gene")

de_36 <- FindMarkers(srt,
  assay = "RNA", ident.1 = "3", ident.2 = "6",
  test.use = "LR",
  only.pos = F,
  min.pct = 0.01
) %>% rownames_to_column("gene")

de_37 <- FindMarkers(srt,
  assay = "RNA", ident.1 = "3", ident.2 = "7",
  test.use = "LR",
  only.pos = F,
  min.pct = 0.01
) %>% rownames_to_column("gene")

de_45 <- FindMarkers(srt,
  assay = "RNA", ident.1 = "4", ident.2 = "5",
  test.use = "LR",
  only.pos = F,
  min.pct = 0.01
) %>% rownames_to_column("gene")

de_46 <- FindMarkers(srt,
  assay = "RNA", ident.1 = "4", ident.2 = "6",
  test.use = "LR",
  only.pos = F,
  min.pct = 0.01
) %>% rownames_to_column("gene")

de_47 <- FindMarkers(srt,
  assay = "RNA", ident.1 = "4", ident.2 = "7",
  test.use = "LR",
  only.pos = F,
  min.pct = 0.01
) %>% rownames_to_column("gene")

de_56 <- FindMarkers(srt,
  assay = "RNA", ident.1 = "5", ident.2 = "6",
  test.use = "LR",
  only.pos = F,
  min.pct = 0.01
) %>% rownames_to_column("gene")

de_57 <- FindMarkers(srt,
  assay = "RNA", ident.1 = "5", ident.2 = "7",
  test.use = "LR",
  only.pos = F,
  min.pct = 0.01
) %>% rownames_to_column("gene")

de_67 <- FindMarkers(srt,
  assay = "RNA", ident.1 = "6", ident.2 = "7",
  test.use = "LR",
  only.pos = F,
  min.pct = 0.01
) %>% rownames_to_column("gene")


o_top_genes <- list()

combinations <- list(
  contrast01 = de_01, contrast02 = de_02, contrast03 = de_03,
  contrast04 = de_04, contrast05 = de_05, contrast06 = de_06,
  contrast07 = de_07, contrast12 = de_12, contrast13 = de_13,
  contrast14 = de_14, contrast15 = de_15, contrast16 = de_16,
  contrast17 = de_17, contrast23 = de_23, contrast24 = de_24,
  contrast25 = de_25, contrast26 = de_26, contrast27 = de_27,
  contrast34 = de_34, contrast35 = de_35, contrast36 = de_36,
  contrast37 = de_37, contrast45 = de_45, contrast46 = de_46,
  contrast47 = de_47, contrast56 = de_56, contrast57 = de_57,
  contrast67 = de_67
) %>% compact()

# Iterate through the 'combinations' list
for (contrast_name in names(combinations)) {
  # Create a unique file name for each data frame
  file_name <- paste0(contrast_name, "_result_origin.csv")

  # Save the data frame as a CSV file
  write_csv(combinations[[contrast_name]], here(tables_dir, file_name))

  # Print a message to indicate progress
  print(paste0("Saved: ", file_name))
}

for (contrast in names(combinations)) {
  top_upregulated <- combinations[[contrast]] %>%
    filter(p_val_adj < 0.05, avg_log2FC > 0, gene %in% gene_int) %>%
    arrange(desc(avg_log2FC)) %>%
    slice_head(n = 5)
  top_downregulated <- combinations[[contrast]] %>%
    filter(p_val_adj < 0.05, avg_log2FC < 0, gene %in% gene_int) %>%
    arrange(avg_log2FC) %>%
    slice_head(n = 5)

  o_top_genes[[contrast]] <- c(top_upregulated$gene, top_downregulated$gene)
}
```

```{r o_collapse-clusters}
DefaultAssay(srt) <- "RNA"
o_cluster.averages <- AverageExpression(srt, return.seurat = TRUE)
DefaultAssay(o_cluster.averages) <- "RNA"

o_cluster.averages
```

```{r de-pairs-plot, fig.width=14, fig.height=13}
plot_pairs <- function(highlight_genes, pair_name) {
  i <- pair_name %>% str_sub(9, 9)
  j <- pair_name %>% str_sub(10, 10)
  print(paste0("Comparison: ", i, " vs ", j))
  plot <- CellScatter(o_cluster.averages, cell1 = i, cell2 = j, features = gene_int[gene_int %in% rownames(o_cluster.averages)], highlight = highlight_genes)
  return(plot)
}
plots <- o_top_genes %>% imap(plot_pairs)
o_plt_grid <- cowplot::plot_grid(plotlist = plots, ncol = 7, labels = names(o_top_genes) %>% map_chr(str_sub, 9, 10), label_size = 8)
o_plt_grid

save_plot(here(plots_dir, "de-pairs-plot_origin.pdf"), o_plt_grid, base_width = 6, base_height = 7, base_asp = 1, ncol = 7, limitsize = FALSE)
```

## Session information

```{r session-info, cache = FALSE}
sI <- sessioninfo::session_info()
sI$loadedOnly <- NULL
print(sI, locale = FALSE)
```